// @vista/web-widgets/lib/widgets/src/components/customer-details-form/types/display-text.js
/**
 * Default display labels for the Customer Details Form widget.
 */
export var defaultCustomerDetailsFormDisplayText = {
    fieldGivenNameLabel: 'Given Name',
    fieldFamilyNameLabel: 'Family Name',
    fieldEmailLabel: 'Email Address',
    fieldPhoneLabel: 'Phone Number',
    fieldGivenNamePlaceholder: undefined,
    fieldFamilyNamePlaceholder: undefined,
    fieldEmailPlaceholder: undefined,
    fieldPhonePlaceholder: undefined,
    errorGivenNameRequiredMessage: 'Please enter your given name.',
    errorGivenNameFormatMessage: 'The given name you entered is invalid.',
    errorGivenNameLengthMessage: 'Your given name must contain less than ${maxLength} characters.',
    errorFamilyNameRequiredMessage: 'Please enter your family name.',
    errorFamilyNameFormatMessage: 'The family name you entered is invalid.',
    errorFamilyNameLengthMessage: 'Your family name must contain less than ${maxLength} characters.',
    errorPhoneRequiredMessage: 'Please enter your phone number.',
    errorPhoneFormatMessage: 'The phone number you entered is invalid.',
    errorPhoneLengthMessage: 'Your phone number must contain less than ${maxLength} characters.',
    errorEmailRequiredMessage: 'Please enter your email address.',
    errorEmailFormatMessage: 'The email address you entered is invalid.',
    errorEmailLengthMessage: 'Your email address must contain less than ${maxLength} characters.',
};
