// @vista/web-widgets/lib/widgets/src/components/film-details/bootstrap.js
import { browsingStoreInstance } from '../../../../api/src/stores/browsing/browsing-store';
import { addWidgetBootstrap } from '../../bootstrap/add-widget-bootstrap';
import { createComponent } from '../../bootstrap/create-component';
import { FilmDetails } from './film-details';
import { mapFilmDetailsAssets } from './mappers/assets-mapper';
import { mapFilmDetailsConfig } from './mappers/config-mapper';
import { mapFilmDetailsDisplayText } from './mappers/display-text-mapper';
/**
 * Renders the film details widget.
 * @param bootstrap
 */
export var filmDetails = function (bootstrap) {
    var filmId = bootstrap.filmId, assets = bootstrap.assets, config = bootstrap.config, displayText = bootstrap.displayText, events = bootstrap.events, selector = bootstrap.selector;
    if (!filmId)
        throw new Error('Cannot render the film details widget without a film id');
    var props = {
        browsingStore: browsingStoreInstance,
        filmId: filmId,
        assets: mapFilmDetailsAssets(assets),
        config: mapFilmDetailsConfig(config),
        displayText: mapFilmDetailsDisplayText(displayText),
        onActionButtonClicked: events ? events.onActionButtonClicked : undefined
    };
    return createComponent(selector, FilmDetails, props);
};
addWidgetBootstrap('filmDetails', filmDetails);
