// @vista/web-widgets/lib/widgets/src/components/film-details/example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { browsingStoreInstance } from '../../../../api/src/stores/browsing/browsing-store';
import { exampleBrowsingStoreData } from '../../../../example-data/browsing/browsing-store-data';
import { exampleFilm } from '../../../../example-data/browsing/film';
import '../../scss/themes/light/film-details.scss';
import { FilmDetailsMessage, FilmDetailsMessageType } from './components/film-details-message';
import { FilmDetailsSkeleton } from './components/film-details-skeleton';
import { FilmDetails } from './film-details';
import { defaultFilmDetailsAssets } from './types/assets';
import { defaultFilmDetailsWidgetConfig } from './types/config';
import { defaultFilmDetailsWidgetDisplayText } from './types/display-text';
var exampleProps = {
    browsingStore: browsingStoreInstance,
    filmId: exampleFilm.id,
    displayText: defaultFilmDetailsWidgetDisplayText,
    config: defaultFilmDetailsWidgetConfig,
    assets: defaultFilmDetailsAssets,
    onActionButtonClicked: function () { return console.log('Action button clicked'); }
};
browsingStoreInstance.setData(exampleBrowsingStoreData);
var FilmDetailsExample = /** @class */ (function (_super) {
    __extends(FilmDetailsExample, _super);
    function FilmDetailsExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FilmDetailsExample.prototype.render = function () {
        return (React.createElement("div", null,
            React.createElement("h1", null, "Film Details"),
            React.createElement("h2", null, "Widget"),
            React.createElement(FilmDetails, __assign({}, exampleProps)),
            React.createElement("h2", null, "Skeleton"),
            React.createElement(FilmDetailsSkeleton, null),
            React.createElement("h2", null, "API Error"),
            React.createElement(FilmDetailsMessage, { displayText: exampleProps.displayText, assets: exampleProps.assets, messageType: FilmDetailsMessageType.FailedToLoadData })));
    };
    return FilmDetailsExample;
}(React.Component));
export { FilmDetailsExample };
