// @vista/web-widgets/lib/widgets/src/dom/key-code.js
/**
 * Represents the key pressed during a KeyboardEvent.
 */
export var KeyCode;
(function (KeyCode) {
    KeyCode[KeyCode["Enter"] = 0] = "Enter";
    KeyCode[KeyCode["Space"] = 1] = "Space";
    KeyCode[KeyCode["Plus"] = 2] = "Plus";
    KeyCode[KeyCode["Minus"] = 3] = "Minus";
    KeyCode[KeyCode["LeftArrow"] = 4] = "LeftArrow";
    KeyCode[KeyCode["UpArrow"] = 5] = "UpArrow";
    KeyCode[KeyCode["RightArrow"] = 6] = "RightArrow";
    KeyCode[KeyCode["DownArrow"] = 7] = "DownArrow";
})(KeyCode || (KeyCode = {}));
/**
 * Gets a recognised KeyCode for a given keyboard event.
 * Returns undefined if the KeyCode is unknown (feel free to add it).
 */
export function getKeyCodeForEvent(event) {
    // tslint:disable-next-line:deprecation Ignore deprecation since no valid cross-browser replacement
    var keyCode = event.which || event.keyCode;
    switch (keyCode) {
        case 13:
            return KeyCode.Enter;
        case 32:
            return KeyCode.Space;
        case 107:
        case 187:
            return KeyCode.Plus;
        case 109:
        case 189:
            return KeyCode.Minus;
        case 37:
            return KeyCode.LeftArrow;
        case 38:
            return KeyCode.UpArrow;
        case 39:
            return KeyCode.RightArrow;
        case 40:
            return KeyCode.DownArrow;
        default:
            return undefined;
    }
}
