// @vista/web-widgets/lib/api/src/widgets-api/sales-data/mappers/tests/test-data.js
import { ItemModelType } from '../../../../../../types/sales-data/item-models/item-model';
export var testMxItemModel = {
    id: '93206984',
    description: 'Whittakers Peanut Slab',
    descriptionTranslations: [],
    extendedDescription: 'The original Whittakers Peanut Slab with 33% cocoa milk chocolate and crunchy roasted peanuts.',
    extendedDescriptionTranslations: [],
    price: 3.80,
};
export var testDomainItemModel = {
    type: ItemModelType.Normal,
    id: '93206984',
    description: 'Whittakers Peanut Slab',
    descriptionTranslations: [],
    shortDescription: undefined,
    extendedDescription: 'The original Whittakers Peanut Slab with 33% cocoa milk chocolate and crunchy roasted peanuts.',
    extendedDescriptionTranslations: [],
    price: 3.80,
    modifierGroups: []
};
export var testMxItemButton = {
    itemModelId: '1234',
    displaySequence: 3
};
export var testDomainItemButton = {
    itemModelId: '1234',
    displaySequence: 3,
    text: undefined
};
export var testMxItemButtonGroup = {
    id: '123',
    title: 'group1',
    titleTranslations: [],
    displaySequence: 2,
    itemButtons: [testMxItemButton],
};
export var testDomainItemButtonGroup = {
    id: '123',
    displaySequence: 2,
    title: 'group1',
    titleTranslations: [],
    itemButtons: [testDomainItemButton],
};
export var testMxGetAllItemButtonGroupsResponseRelatedData = {
    itemModels: [testMxItemModel]
};
export var testDomainGetAllItemButtonGroupsResponseRelatedData = {
    itemModels: [testDomainItemModel]
};
export var testMxGetAllItemButtonGroupsResponse = {
    itemButtonGroups: [testMxItemButtonGroup],
    relatedData: testMxGetAllItemButtonGroupsResponseRelatedData
};
export var testDomainGetAllItemButtonGroupsResponse = {
    itemButtonGroups: [testDomainItemButtonGroup],
    relatedData: testDomainGetAllItemButtonGroupsResponseRelatedData
};
