// @vista/web-widgets/lib/widgets/src/components/showtime-details/example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import '../../scss/themes/light/_all.scss';
import { ShowtimeDetails } from './showtime-details';
import { defaultShowtimeDetailsDisplayText } from './showtime-details-display-text';
var showtimeDetailsProps = {
    filmTitle: 'Gone With the Wind',
    startTime: new Date('11 Jan 2019'),
    siteName: 'ABC Cinemas Queen Street',
    siteAddress: '123 Queen Street, Auckland 1010',
    displayText: defaultShowtimeDetailsDisplayText
};
var ShowtimeDetailsExample = /** @class */ (function (_super) {
    __extends(ShowtimeDetailsExample, _super);
    function ShowtimeDetailsExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ShowtimeDetailsExample.prototype.render = function () {
        return (React.createElement("div", { style: { margin: '40px' } },
            React.createElement(ShowtimeDetails, __assign({}, showtimeDetailsProps))));
    };
    return ShowtimeDetailsExample;
}(React.Component));
export { ShowtimeDetailsExample };
