// @vista/web-widgets/lib/api/src/mappers/sales-data/ticket-type-mapper.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { TicketTypeCategory } from '../../../../types/sales-data/ticket-type';
var mapToWidgetTicketType = function (mxTicketType, baseTicketType) {
    switch (mxTicketType.category) {
        case 'Voucher': return __assign({}, baseTicketType, { category: TicketTypeCategory.Voucher });
        case 'Loyalty': return __assign({}, baseTicketType, { category: TicketTypeCategory.Loyalty, rewardDetails: mxTicketType.rewardDetails });
        case 'Package': return __assign({}, baseTicketType, { category: TicketTypeCategory.Package, packageContents: mxTicketType.packageContent });
        case 'Standard': return __assign({}, baseTicketType, { category: TicketTypeCategory.Standard });
    }
};
var getKeyForMxVoucherTicketType = function (ticketType) {
    return "voucher-ticket-type-" + ticketType.id;
};
var getKeyForMxPackageTicketType = function (ticketType) {
    return "package-ticket-type-" + ticketType.id;
};
var getKeyForMxLoyaltyTicketType = function (ticketType) {
    return "loyalty-ticket-type-" + ticketType.id + "-reward-" + ticketType.rewardDetails.id;
};
var getKeyForMxStandardTicketType = function (ticketType) {
    return "standard-ticket-type-" + ticketType.id;
};
/**
 * Generates a unique key for a mx ticket type.
 */
var getKeyForMxTicketType = function (mxTicketType) {
    switch (mxTicketType.category) {
        case 'Voucher': return getKeyForMxVoucherTicketType(mxTicketType);
        case 'Package': return getKeyForMxPackageTicketType(mxTicketType);
        case 'Loyalty': return getKeyForMxLoyaltyTicketType(mxTicketType);
        default: return getKeyForMxStandardTicketType(mxTicketType);
    }
};
export var mapMxTicketTypeToWidgetTicketType = function (mxTicketType) {
    var baseTicketType = {
        id: mxTicketType.id,
        key: getKeyForMxTicketType(mxTicketType),
        description: mxTicketType.description,
        descriptionTranslations: mxTicketType.descriptionTranslations,
        areaCategoryCode: mxTicketType.areaCategoryCode,
        price: mxTicketType.price,
        category: TicketTypeCategory.Standard
    };
    return mapToWidgetTicketType(mxTicketType, baseTicketType);
};
export var mapMxTicketTypesToWidgetTicketTypes = function (mxTicketTypes) {
    return mxTicketTypes.map(mapMxTicketTypeToWidgetTicketType);
};
