// @vista/web-widgets/lib/widgets/src/components/site-list/bootstrap.js
import { browsingStoreInstance } from '../../../../api/src/stores/browsing/browsing-store';
import { configurationStoreInstance } from '../../../../api/src/stores/configuration/configuration-store';
import { addWidgetBootstrap } from '../../bootstrap/add-widget-bootstrap';
import { createComponent } from '../../bootstrap/create-component';
import { mapSiteListAssets } from './mappers/assets-mapper';
import { mapSiteListDisplayText } from './mappers/display-text-mapper';
import { SiteList } from './site-list';
/**
 * Renders the Site List widget.
 * @param bootstrap
 */
export var siteList = function (bootstrap) {
    var selector = bootstrap.selector, displayText = bootstrap.displayText, assets = bootstrap.assets, events = bootstrap.events;
    var props = {
        browsingStore: browsingStoreInstance,
        configurationStore: configurationStoreInstance,
        displayText: mapSiteListDisplayText(displayText),
        assets: mapSiteListAssets(assets),
        onSiteClicked: events ? events.onSiteClicked : undefined
    };
    return createComponent(selector, SiteList, props);
};
addWidgetBootstrap('siteList', siteList);
