// @vista/web-widgets/lib/widgets/src/components/site-list/example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { browsingStoreInstance } from '../../../../api/src/stores/browsing/browsing-store';
import { configurationStoreInstance } from '../../../../api/src/stores/configuration/configuration-store';
import { exampleSites } from '../../../../example-data/browsing/site';
// import '../../scss/themes/dark/site-list.scss';
import '../../scss/themes/light/site-list.scss';
import { SiteListMessage, SiteListMessageType } from './components/site-list-message';
import { SiteListSkeleton } from './components/site-list-skeleton';
import { SiteList } from './site-list';
import { defaultSiteListAssets } from './types/assets';
import { defaultSiteListDisplayText } from './types/display-text';
var exampleProps = {
    browsingStore: browsingStoreInstance,
    configurationStore: configurationStoreInstance,
    displayText: defaultSiteListDisplayText,
    assets: defaultSiteListAssets,
    onSiteClicked: function (site) { return console.log('Site clicked', site.id); }
};
var testSiteListMessageProps = {
    messageType: SiteListMessageType.FailedToLoadData,
    displayText: defaultSiteListDisplayText,
    assets: defaultSiteListAssets
};
browsingStoreInstance.actions.setSites(exampleSites);
setTimeout(function () { return browsingStoreInstance.setData({ allSitesLoaded: true }); }, 3000);
var SiteListExample = /** @class */ (function (_super) {
    __extends(SiteListExample, _super);
    function SiteListExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    SiteListExample.prototype.render = function () {
        return (React.createElement("div", { style: { padding: '20px' } },
            React.createElement("h1", null, "Site list"),
            React.createElement(SiteList, __assign({}, exampleProps)),
            React.createElement("br", null),
            React.createElement("h2", null, "Skeleton"),
            React.createElement(SiteListSkeleton, null),
            React.createElement("br", null),
            React.createElement("h2", null, "Message"),
            React.createElement(SiteListMessage, __assign({}, testSiteListMessageProps))));
    };
    return SiteListExample;
}(React.Component));
export { SiteListExample };
