// @vista/web-widgets/lib/widgets/src/components/customer-details-form/example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { orderStoreInstance } from '../../../../api/src/stores/order/order-store';
import { exampleOrderCustomerDetails } from '../../../../example-data/order/order-customer-details';
import { defaultConfigurationStoreData } from '../../../../types/stores/configuration-store-data';
// import '../../scss/themes/dark/customer-details-form.scss';
import '../../scss/themes/light/customer-details-form.scss';
import { Button, ButtonColor, ButtonSize } from '../button/button';
import { CustomerDetailsForm } from './customer-details-form';
import { mapCustomerDetailsFormConfig } from './mappers/config-mapper';
import { defaultCustomerDetailsFormDisplayText } from './types/display-text';
var orderStore = orderStoreInstance;
orderStore.setData({ customerDetails: exampleOrderCustomerDetails });
var exampleCustomerDetailsProps = {
    orderStore: orderStore,
    validationConfig: defaultConfigurationStoreData.customerDetails.validation,
    config: mapCustomerDetailsFormConfig(),
    displayText: defaultCustomerDetailsFormDisplayText,
    onSubmit: function (customerDetails) { console.log(customerDetails); }
};
var CustomerDetailsFormExample = /** @class */ (function (_super) {
    __extends(CustomerDetailsFormExample, _super);
    function CustomerDetailsFormExample() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.customerDetailsFormRef = React.createRef();
        _this.validateForm = function () {
            var customerDetailsForm = _this.customerDetailsFormRef.current;
            if (customerDetailsForm)
                customerDetailsForm.attemptSubmit().catch();
        };
        return _this;
    }
    CustomerDetailsFormExample.prototype.render = function () {
        return (React.createElement("div", { style: { display: 'block', margin: '30px auto', maxWidth: '500px' } },
            React.createElement("h1", null, "Customer details form"),
            React.createElement(CustomerDetailsForm, __assign({}, exampleCustomerDetailsProps, { ref: this.customerDetailsFormRef })),
            React.createElement("br", null),
            React.createElement(Button, { color: ButtonColor.Primary, size: ButtonSize.Large, label: "Validate Details", onClick: this.validateForm })));
    };
    return CustomerDetailsFormExample;
}(React.Component));
export { CustomerDetailsFormExample };
