// @vista/web-widgets/lib/widgets/src/components/site-details/example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { browsingStoreInstance } from '../../../../api/src/stores/browsing/browsing-store';
import { exampleSite } from '../../../../example-data/browsing/site';
// import '../../scss/themes/dark/site-details.scss';
import '../../scss/themes/light/site-details.scss';
import { SiteDetails } from './site-details';
import { SiteDetailsMessage, SiteDetailsMessageType } from './components/site-details-message';
import { SiteDetailsSkeleton } from './components/site-details-skeleton';
import { defaultSiteDetailsAssets } from './types/assets';
import { defaultSiteDetailsWidgetDisplayText } from './types/display-text';
var exampleProps = {
    browsingStore: browsingStoreInstance,
    siteId: exampleSite.id,
    displayText: defaultSiteDetailsWidgetDisplayText,
    assets: defaultSiteDetailsAssets
};
browsingStoreInstance.actions.addSites([exampleSite]);
var SiteDetailsExample = /** @class */ (function (_super) {
    __extends(SiteDetailsExample, _super);
    function SiteDetailsExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    SiteDetailsExample.prototype.render = function () {
        return (React.createElement("div", null,
            React.createElement("h1", null, "Site Details"),
            React.createElement("h2", null, "Widget"),
            React.createElement(SiteDetails, __assign({}, exampleProps)),
            React.createElement("h2", null, "Skeleton"),
            React.createElement(SiteDetailsSkeleton, null),
            React.createElement("h2", null, "Message"),
            React.createElement(SiteDetailsMessage, { messageType: SiteDetailsMessageType.FailedToLoadData, displayText: defaultSiteDetailsWidgetDisplayText, assets: defaultSiteDetailsAssets })));
    };
    return SiteDetailsExample;
}(React.Component));
export { SiteDetailsExample };
