// @vista/web-widgets/lib/widgets/src/components/seat-picker/components/seat-picker-message.js
import * as React from 'react';
import { Message } from '../../message/message';
export var SeatPickerMessageType;
(function (SeatPickerMessageType) {
    SeatPickerMessageType["FailedToLoadData"] = "FailedToLoadData";
})(SeatPickerMessageType || (SeatPickerMessageType = {}));
function getIconName(messageType, assets) {
    switch (messageType) {
        case SeatPickerMessageType.FailedToLoadData:
            return assets.failedToLoadDataIconName;
    }
}
function getTitleString(messageType, displayText) {
    switch (messageType) {
        case SeatPickerMessageType.FailedToLoadData:
            return displayText.failedToLoadDataTitle;
        default: return undefined;
    }
}
function getDescriptionString(messageType, displayText) {
    switch (messageType) {
        case SeatPickerMessageType.FailedToLoadData:
            return displayText.failedToLoadDataMessage;
        default: return undefined;
    }
}
export var SeatPickerMessage = function (_a) {
    var messageType = _a.messageType, displayText = _a.displayText, assets = _a.assets;
    return (React.createElement(Message, { icon: getIconName(messageType, assets), title: getTitleString(messageType, displayText), description: getDescriptionString(messageType, displayText), className: 'v-seat-picker-message' }));
};
