// @vista/web-widgets/lib/widgets/src/components/member-transaction-history/components/member-transaction-history-message.js
import * as React from 'react';
import { Message } from '../../message/message';
export var MemberTransactionHistoryMessageType;
(function (MemberTransactionHistoryMessageType) {
    MemberTransactionHistoryMessageType["NoMemberTransactionData"] = "NoMemberTransactionData";
    MemberTransactionHistoryMessageType["FailedToLoadData"] = "FailedToLoadData";
})(MemberTransactionHistoryMessageType || (MemberTransactionHistoryMessageType = {}));
function getIconName(messageType, assets) {
    switch (messageType) {
        case MemberTransactionHistoryMessageType.NoMemberTransactionData:
            return assets.noTransactionHistoryIconName;
        case MemberTransactionHistoryMessageType.FailedToLoadData:
            return assets.failedToLoadDataIconName;
        default:
            return assets.noTransactionHistoryIconName;
    }
}
function getTitleString(messageType, displayText) {
    switch (messageType) {
        case MemberTransactionHistoryMessageType.NoMemberTransactionData:
            return displayText.noMemberTransactionHistoryTitle;
        case MemberTransactionHistoryMessageType.FailedToLoadData:
            return displayText.failedToLoadDataTitle;
        default:
            return undefined;
    }
}
function getDescriptionString(messageType, displayText) {
    switch (messageType) {
        case MemberTransactionHistoryMessageType.NoMemberTransactionData:
            return displayText.noMemberTransactionHistoryMessage;
        case MemberTransactionHistoryMessageType.FailedToLoadData:
            return displayText.failedToLoadDataMessage;
        default:
            return undefined;
    }
}
export var MemberTransactionHistoryMessage = function (_a) {
    var messageType = _a.messageType, displayText = _a.displayText, assets = _a.assets;
    return (React.createElement(Message, { icon: getIconName(messageType, assets), title: getTitleString(messageType, displayText), description: getDescriptionString(messageType, displayText), className: 'v-member-transaction-history-message' }));
};
