// @vista/web-widgets/lib/api/src/stores/seating/helpers/rules/seat-selection-rule.js
/**
 * Seat selection rule names
 */
export var SeatSelectionRuleName;
(function (SeatSelectionRuleName) {
    SeatSelectionRuleName[SeatSelectionRuleName["CannotExceedAllocatedSeatsForAreaCategory"] = 0] = "CannotExceedAllocatedSeatsForAreaCategory";
    SeatSelectionRuleName[SeatSelectionRuleName["CannotLeaveGapsBetweenSelectedSeats"] = 1] = "CannotLeaveGapsBetweenSelectedSeats";
    SeatSelectionRuleName[SeatSelectionRuleName["CannotLeaveSingleSeatGapFromAisle"] = 2] = "CannotLeaveSingleSeatGapFromAisle";
    SeatSelectionRuleName[SeatSelectionRuleName["CannotLeaveSingleSeatGapFromUnavailableSeat"] = 3] = "CannotLeaveSingleSeatGapFromUnavailableSeat";
    SeatSelectionRuleName[SeatSelectionRuleName["CannotSelectUnavailableSeats"] = 4] = "CannotSelectUnavailableSeats";
    SeatSelectionRuleName[SeatSelectionRuleName["MustFillSofaSeats"] = 5] = "MustFillSofaSeats";
    SeatSelectionRuleName[SeatSelectionRuleName["MustSelectAllSeatsAllocatedForOrder"] = 6] = "MustSelectAllSeatsAllocatedForOrder";
    SeatSelectionRuleName[SeatSelectionRuleName["MustSelectAssociatedWheelchairSeat"] = 7] = "MustSelectAssociatedWheelchairSeat";
    SeatSelectionRuleName[SeatSelectionRuleName["MustSelectAtLeastOneSeat"] = 8] = "MustSelectAtLeastOneSeat";
})(SeatSelectionRuleName || (SeatSelectionRuleName = {}));
