// @vista/web-widgets/lib/example-data/sales-data/item-button-basic.js
import * as BasicItemModels from './item-model-basic';
export var exampleBasicCandyItemButton1 = {
    itemModelId: BasicItemModels.exampleBasicCandyItemModel1.id,
    displaySequence: 1
};
export var exampleBasicCandyItemButton2 = {
    itemModelId: BasicItemModels.exampleBasicCandyItemModel2.id,
    displaySequence: 2
};
export var exampleBasicCandyItemButton3 = {
    itemModelId: BasicItemModels.exampleBasicCandyItemModel3.id,
    displaySequence: 3
};
export var exampleBasicCandyItemButton4 = {
    itemModelId: BasicItemModels.exampleBasicCandyItemModel4.id,
    displaySequence: 4
};
export var exampleBasicCandyItemButton5 = {
    itemModelId: BasicItemModels.exampleBasicCandyItemModel5.id,
    displaySequence: 5
};
export var exampleBasicDrinkItemButton1 = {
    itemModelId: BasicItemModels.exampleBasicDrinkItemModel1.id,
    displaySequence: 1
};
export var exampleBasicDrinkItemButton2 = {
    itemModelId: BasicItemModels.exampleBasicDrinkItemModel2.id,
    displaySequence: 2
};
export var exampleBasicDrinkItemButton3 = {
    itemModelId: BasicItemModels.exampleBasicDrinkItemModel3.id,
    displaySequence: 3
};
export var exampleBasicDrinkItemButton4 = {
    itemModelId: BasicItemModels.exampleBasicDrinkItemModel4.id,
    displaySequence: 4
};
export var exampleBasicDrinkItemButton5 = {
    itemModelId: BasicItemModels.exampleBasicDrinkItemModel5.id,
    displaySequence: 5
};
