// @vista/web-widgets/lib/example-data/sales-data/item-button-group.js
import * as BasicItemButtons from './item-button-basic';
export var exampleBasicItemButtonGroup1 = {
    id: '1',
    displaySequence: 1,
    title: 'Candy',
    itemButtons: [
        BasicItemButtons.exampleBasicCandyItemButton1,
        BasicItemButtons.exampleBasicCandyItemButton2,
        BasicItemButtons.exampleBasicCandyItemButton3,
        BasicItemButtons.exampleBasicCandyItemButton4,
        BasicItemButtons.exampleBasicCandyItemButton5
    ],
    titleTranslations: []
};
export var exampleBasicItemButtonGroup2 = {
    id: '2',
    displaySequence: 2,
    title: 'Soft Drinks',
    itemButtons: [
        BasicItemButtons.exampleBasicDrinkItemButton1,
        BasicItemButtons.exampleBasicDrinkItemButton2,
        BasicItemButtons.exampleBasicDrinkItemButton3,
        BasicItemButtons.exampleBasicDrinkItemButton4,
        BasicItemButtons.exampleBasicDrinkItemButton5
    ],
    titleTranslations: []
};
export var exampleItemButtonGroups = [
    exampleBasicItemButtonGroup1,
    exampleBasicItemButtonGroup2
];
