// @vista/web-widgets/lib/example-data/sales-data/item-model-basic.js
import { ItemModelType } from '../../types/sales-data/item-models/item-model';
// BASIC ITEMS 1
export var exampleBasicCandyItemModel1 = {
    type: ItemModelType.Normal,
    id: '93206984',
    description: 'Whittaker\'s Peanut Slab',
    shortDescription: 'peanutslab',
    extendedDescription: 'The original Whittaker\'s Peanut Slab with 33% cocoa milk chocolate and crunchy roasted peanuts.',
    price: 3.80,
    modifierGroups: [],
    descriptionTranslations: [],
    extendedDescriptionTranslations: []
};
export var exampleBasicCandyItemModel2 = {
    type: ItemModelType.Normal,
    id: '93206987',
    description: 'Whittaker\'s Hokey Pokey Slab',
    shortDescription: 'hokeypokey',
    extendedDescription: 'The classic kiwi taste of crunchy honeycomb in 5 Roll Refined Creamy Milk chocolate.',
    price: 3.80,
    modifierGroups: [],
    descriptionTranslations: [],
    extendedDescriptionTranslations: []
};
export var exampleBasicCandyItemModel3 = {
    type: ItemModelType.Normal,
    id: '93206986',
    description: 'Whittaker\'s Creamy Milk Slab',
    shortDescription: 'peanutslab',
    extendedDescription: '33% cocoa, super smooth 5 Roll Refined Creamy Milk chocolate in a 50g slab.',
    price: 3.80,
    modifierGroups: [],
    descriptionTranslations: [],
    extendedDescriptionTranslations: []
};
export var exampleBasicCandyItemModel4 = {
    type: ItemModelType.Normal,
    id: '93206992',
    description: 'Whittaker\'s Orange K Bar',
    shortDescription: 'kbaro',
    extendedDescription: undefined,
    price: 1.50,
    modifierGroups: [],
    descriptionTranslations: [],
    extendedDescriptionTranslations: []
};
export var exampleBasicCandyItemModel5 = {
    type: ItemModelType.Normal,
    id: '93206990',
    description: 'Whittaker\'s Raspberry K Bar',
    shortDescription: 'kbarr',
    extendedDescription: undefined,
    price: 1.50,
    modifierGroups: [],
    descriptionTranslations: [],
    extendedDescriptionTranslations: []
};
export var exampleBasicDrinkItemModel1 = {
    type: ItemModelType.Normal,
    id: '93206964',
    description: 'Coca Cola (420ml)',
    shortDescription: 'coke',
    price: 3.5,
    modifierGroups: [],
    descriptionTranslations: [],
    extendedDescriptionTranslations: []
};
export var exampleBasicDrinkItemModel2 = {
    type: ItemModelType.Normal,
    id: '93206965',
    description: 'Coke No Sugar (420ml)',
    shortDescription: 'cokens',
    price: 3.5,
    modifierGroups: [],
    descriptionTranslations: [],
    extendedDescriptionTranslations: []
};
export var exampleBasicDrinkItemModel3 = {
    type: ItemModelType.Normal,
    id: '93206967',
    description: 'Sprite (420ml)',
    shortDescription: 'sprite',
    price: 3.5,
    modifierGroups: [],
    descriptionTranslations: [],
    extendedDescriptionTranslations: []
};
export var exampleBasicDrinkItemModel4 = {
    type: ItemModelType.Normal,
    id: '93206968',
    description: 'Fanta (420ml)',
    shortDescription: 'fanta',
    price: 3.5,
    modifierGroups: [],
    descriptionTranslations: [],
    extendedDescriptionTranslations: []
};
export var exampleBasicDrinkItemModel5 = {
    type: ItemModelType.Normal,
    id: '93206973',
    description: 'Red Bull (250ml)',
    shortDescription: 'redbull',
    price: 4.0,
    modifierGroups: [],
    descriptionTranslations: [],
    extendedDescriptionTranslations: []
};
export var exampleBasicItemModels = [
    exampleBasicCandyItemModel1,
    exampleBasicCandyItemModel2,
    exampleBasicCandyItemModel3,
    exampleBasicCandyItemModel4,
    exampleBasicCandyItemModel5,
    exampleBasicDrinkItemModel1,
    exampleBasicDrinkItemModel2,
    exampleBasicDrinkItemModel3,
    exampleBasicDrinkItemModel4,
    exampleBasicDrinkItemModel5
];
