// @vista/web-widgets/lib/mx-api-client/src/mxt/api/mxt-api.js
import { cancelOrder } from './cancel-order';
import { createOrder } from './create-order';
import { createOrderWithSeats } from './create-order-with-seats';
import { getJourneyById } from './get-journey-by-id';
import { getJourneyByReference } from './get-journey-by-reference';
import { getOrderCompletionStatus } from './get-order-completion-status';
import { getSeatLayoutForOrder } from './get-seat-layout-for-order';
import { getSeatLayoutForShowtime } from './get-seat-layout-for-showtime';
import { setCustomerDetails } from './set-customer-details';
import { setOrderItems } from './set-order-items';
import { startEmbeddedPayment } from './start-embedded-payment';
import { updateSelectedSeats } from './update-selected-seats';
import { updateShowtimeTicketingDetails } from './update-showtime-ticketing-details';
/**
 * @hidden
 */
export var mxtApi = {
    createOrder: createOrder,
    createOrderWithSeats: createOrderWithSeats,
    cancelOrder: cancelOrder,
    setCustomerDetails: setCustomerDetails,
    getSeatLayoutForOrder: getSeatLayoutForOrder,
    getSeatLayoutForShowtime: getSeatLayoutForShowtime,
    updateShowtimeTicketingDetails: updateShowtimeTicketingDetails,
    updateSelectedSeats: updateSelectedSeats,
    setOrderItems: setOrderItems,
    startEmbeddedPayment: startEmbeddedPayment,
    getOrderCompletionStatus: getOrderCompletionStatus,
    getJourneyByReference: getJourneyByReference,
    getJourneyById: getJourneyById
};
