// @vista/web-widgets/lib/widgets/src/components/film-details/components/film-details-message.js
import * as React from 'react';
import { Message } from '../../message/message';
export var FilmDetailsMessageType;
(function (FilmDetailsMessageType) {
    FilmDetailsMessageType["FailedToLoadData"] = "FailedToLoadData";
})(FilmDetailsMessageType || (FilmDetailsMessageType = {}));
function getIconName(messageType, assets) {
    switch (messageType) {
        case FilmDetailsMessageType.FailedToLoadData:
            return assets.failedToLoadDataIconName;
    }
}
function getTitleString(messageType, displayText) {
    switch (messageType) {
        case FilmDetailsMessageType.FailedToLoadData:
            return displayText.failedToLoadDataTitle;
        default: return undefined;
    }
}
function getDescriptionString(messageType, displayText) {
    switch (messageType) {
        case FilmDetailsMessageType.FailedToLoadData:
            return displayText.failedToLoadDataMessage;
        default: return undefined;
    }
}
export var FilmDetailsMessage = function (_a) {
    var messageType = _a.messageType, displayText = _a.displayText, assets = _a.assets;
    return (React.createElement(Message, { icon: getIconName(messageType, assets), title: getTitleString(messageType, displayText), description: getDescriptionString(messageType, displayText), className: 'v-film-details-message' }));
};
