// @vista/web-widgets/lib/widgets/src/components/ticket-picker/example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { orderService } from '../../../../api/src/services/order/order-service';
import { configurationStoreInstance } from '../../../../api/src/stores/configuration/configuration-store';
import { orderStoreInstance } from '../../../../api/src/stores/order/order-store';
import { browserSessionServiceInstance } from '../../../../api/src/user/browser-session-service';
import { exampleOrder } from '../../../../example-data/order/order';
import { exampleOrderShowtime } from '../../../../example-data/order/order-showtime';
// import '../../scss/themes/dark/ticket-picker.scss';
import '../../scss/themes/light/ticket-picker.scss';
import { Button, ButtonColor, ButtonSize } from '../button/button';
import { TicketPickerMessage, TicketPickerMessageType } from './components/ticket-picker-message';
import { TicketPickerSkeleton } from './components/ticket-picker-skeleton';
import { TicketPicker } from './ticket-picker';
import { defaultTicketPickerAssets } from './types/assets';
import { defaultTicketPickerConfig } from './types/config';
import { defaultTicketPickerDisplayText } from './types/display-text';
var showtimeId = exampleOrderShowtime.showtimeId;
window.vista = {
    stores: {
        order: orderStoreInstance,
        configuration: configurationStoreInstance
    }
};
orderStoreInstance.actions.setOrderData(__assign({}, exampleOrder, { bookingFee: 1 }));
var TicketPickerExample = /** @class */ (function (_super) {
    __extends(TicketPickerExample, _super);
    function TicketPickerExample(props) {
        var _this = _super.call(this, props) || this;
        browserSessionServiceInstance.start();
        return _this;
    }
    TicketPickerExample.prototype.validateSelection = function () {
        orderStoreInstance.actions.validateSelectedTicketsForShowtime(showtimeId);
    };
    TicketPickerExample.prototype.render = function () {
        return (React.createElement("div", null,
            React.createElement("h1", null, "Ticket Picker"),
            React.createElement(TicketPicker, { orderStore: orderStoreInstance, orderService: orderService, showtimeId: showtimeId, config: defaultTicketPickerConfig, displayText: defaultTicketPickerDisplayText, assets: defaultTicketPickerAssets }),
            React.createElement("br", null),
            React.createElement(Button, { color: ButtonColor.Primary, size: ButtonSize.Large, label: "Validate Selection", onClick: this.validateSelection }),
            React.createElement("br", null),
            React.createElement(TicketPickerSkeleton, null),
            React.createElement("br", null),
            React.createElement(TicketPickerMessage, { messageType: TicketPickerMessageType.FailedToLoadData, displayText: defaultTicketPickerDisplayText, assets: defaultTicketPickerAssets })));
    };
    return TicketPickerExample;
}(React.Component));
export { TicketPickerExample };
