// @vista/web-widgets/lib/example-data/browsing/browsing-store-data.js
import { convertArrayToMap } from '../../widgets/src/utils/array-utils';
import { exampleCensorRatings } from './censor-rating';
import { exampleFilms } from './film';
import { exampleGenres } from './genre';
import { examplePeople } from './person';
import { exampleShowtimeAttributes } from './showtime-attributes';
import { exampleSite, exampleSites } from './site';
export var exampleBrowsingStoreData = {
    filmsById: convertArrayToMap(exampleFilms, 'id'),
    showtimesByQuery: {},
    showtimesById: new Map(),
    sitesById: convertArrayToMap(exampleSites, 'id'),
    showtimeAttributesById: convertArrayToMap(exampleShowtimeAttributes, 'id'),
    censorRatingsById: convertArrayToMap(exampleCensorRatings, 'id'),
    genresById: convertArrayToMap(exampleGenres, 'id'),
    peopleById: convertArrayToMap(examplePeople, 'id'),
    screeningDates: undefined,
    selectedSiteIds: [exampleSites[0].id],
    allFilmsLoaded: true,
    allSitesLoaded: true,
    screeningDatesByQuery: {},
    filmsBySiteId: new Map([[exampleSite.id, exampleFilms]])
};
