// @vista/web-widgets/lib/example-data/browsing/censor-rating.js
export var ExampleCensorRatings;
(function (ExampleCensorRatings) {
    ExampleCensorRatings.M = {
        id: '1',
        classification: 'M',
        classificationTranslations: [],
        description: 'Rated M',
        descriptionTranslations: []
    };
    ExampleCensorRatings.R13 = {
        id: '2',
        classification: 'R13',
        classificationTranslations: [],
        description: 'Recommended 13 years and older',
        descriptionTranslations: []
    };
    ExampleCensorRatings.PG = {
        id: '3',
        classification: 'PG',
        classificationTranslations: [],
        description: 'Parental guidance is recommended',
        descriptionTranslations: []
    };
    ExampleCensorRatings.G = {
        id: '4',
        classification: 'G',
        classificationTranslations: [],
        description: 'Suitable for all audiences',
        descriptionTranslations: []
    };
})(ExampleCensorRatings || (ExampleCensorRatings = {}));
export var exampleCensorRating = ExampleCensorRatings.G;
export var exampleCensorRatings = [
    ExampleCensorRatings.M,
    ExampleCensorRatings.R13,
    ExampleCensorRatings.PG,
    ExampleCensorRatings.G
];
