// @vista/web-widgets/lib/example-data/browsing/film.js
import { FilmCategory } from '../../types/browsing/film-category';
import { ExampleCensorRatings } from './censor-rating';
import { ExampleGenres } from './genre';
import { ExamplePeople } from './person';
export var exampleFilms = [
    {
        id: '1',
        title: 'Star Wars: The Last Jedi',
        titleTranslations: [
            {
                languageTag: 'en',
                text: 'Star Wars: The Last Jedi'
            },
            {
                languageTag: 'fr-CA',
                text: 'Star Wars: Les Derniers Jedi'
            }
        ],
        plot: 'Having taken her first steps into a larger world in Star Wars: The Force Awakens (2015), Rey continues her epic journey with Finn, Poe, and Luke Skywalker in the next chapter of the saga.',
        plotTranslations: [],
        classificationNote: 'Contains violence and offensive language.',
        classificationNoteTranslations: [],
        censorRatingId: ExampleCensorRatings.M.id,
        releaseDate: new Date('28 Apr 2018'),
        runtime: 152,
        displayPriority: 1,
        actorIds: [
            ExamplePeople.MarkHamill.id,
            ExamplePeople.CarrieFisher.id,
            ExamplePeople.AdamDriver.id,
            ExamplePeople.DaisyRidley.id,
            ExamplePeople.JohnBoyega.id
        ],
        directorIds: [ExamplePeople.RianJohnson.id],
        genreIds: [ExampleGenres.Action.id],
        categories: [FilmCategory.NowShowing]
    },
    {
        id: '2',
        title: 'Chasing Great',
        titleTranslations: [],
        plotTranslations: [],
        classificationNoteTranslations: [],
        censorRatingId: ExampleCensorRatings.M.id,
        runtime: 120,
        displayPriority: 2,
        actorIds: [
            ExamplePeople.TomHanks.id,
            ExamplePeople.AaronEckhart.id,
            ExamplePeople.LauraLinney.id,
            ExamplePeople.AnnaGunn.id,
            ExamplePeople.AutumnReeser.id
        ],
        directorIds: [ExamplePeople.ClintEastwood.id],
        genreIds: [],
        categories: [FilmCategory.NowShowing, FilmCategory.ComingSoon]
    },
    {
        id: '3',
        title: 'Blade Runner 2049',
        titleTranslations: [],
        plotTranslations: [],
        classificationNoteTranslations: [],
        censorRatingId: ExampleCensorRatings.R13.id,
        runtime: 163,
        displayPriority: 2,
        actorIds: [
            ExamplePeople.RyanGosling.id,
            ExamplePeople.CarrieFisher.id,
            ExamplePeople.JohnBoyega.id
        ],
        directorIds: [
            ExamplePeople.TomHanks.id
        ],
        genreIds: [],
        categories: [FilmCategory.NowShowing, FilmCategory.ComingSoon]
    },
    {
        id: '4',
        title: 'Daddy\'s Home 2',
        titleTranslations: [],
        plotTranslations: [],
        classificationNoteTranslations: [],
        runtime: 98,
        censorRatingId: ExampleCensorRatings.PG.id,
        displayPriority: 2,
        actorIds: [
            ExamplePeople.RyanGosling.id,
            ExamplePeople.CarrieFisher.id,
            ExamplePeople.JohnBoyega.id
        ],
        directorIds: [
            ExamplePeople.TomHanks.id
        ],
        genreIds: [],
        categories: [FilmCategory.NowShowing, FilmCategory.ComingSoon]
    },
    {
        id: '5',
        title: 'Venom',
        titleTranslations: [],
        plotTranslations: [],
        classificationNoteTranslations: [],
        censorRatingId: ExampleCensorRatings.M.id,
        runtime: 98,
        displayPriority: 2,
        actorIds: [
            ExamplePeople.RyanGosling.id,
            ExamplePeople.CarrieFisher.id,
            ExamplePeople.JohnBoyega.id
        ],
        directorIds: [
            ExamplePeople.TomHanks.id
        ],
        genreIds: [],
        categories: [FilmCategory.NowShowing, FilmCategory.ComingSoon]
    },
    {
        id: '6',
        title: 'First Man',
        titleTranslations: [],
        plotTranslations: [],
        classificationNoteTranslations: [],
        censorRatingId: ExampleCensorRatings.M.id,
        runtime: 98,
        displayPriority: 2,
        actorIds: [
            ExamplePeople.RyanGosling.id,
            ExamplePeople.CarrieFisher.id,
            ExamplePeople.JohnBoyega.id
        ],
        directorIds: [
            ExamplePeople.TomHanks.id
        ],
        genreIds: [],
        categories: [FilmCategory.NowShowing, FilmCategory.ComingSoon]
    },
    {
        id: '7',
        title: 'Goosebump 2: Haunted Halloween',
        titleTranslations: [],
        plotTranslations: [],
        classificationNoteTranslations: [],
        censorRatingId: ExampleCensorRatings.PG.id,
        runtime: 98,
        displayPriority: 2,
        actorIds: [
            ExamplePeople.RyanGosling.id,
            ExamplePeople.CarrieFisher.id,
            ExamplePeople.JohnBoyega.id
        ],
        directorIds: [
            ExamplePeople.TomHanks.id
        ],
        genreIds: [],
        categories: [FilmCategory.NowShowing, FilmCategory.ComingSoon]
    },
    {
        id: '8',
        title: 'Johnny English Strikes Again',
        titleTranslations: [],
        plotTranslations: [],
        classificationNoteTranslations: [],
        censorRatingId: ExampleCensorRatings.G.id,
        runtime: 98,
        displayPriority: 2,
        actorIds: [
            ExamplePeople.RyanGosling.id,
            ExamplePeople.CarrieFisher.id,
            ExamplePeople.JohnBoyega.id
        ],
        directorIds: [
            ExamplePeople.TomHanks.id
        ],
        genreIds: [ExampleGenres.Drama.id, ExampleGenres.Comedy.id],
        categories: [FilmCategory.NowShowing, FilmCategory.ComingSoon]
    },
    {
        id: '9',
        title: 'Christopher Robin',
        titleTranslations: [],
        plotTranslations: [],
        classificationNoteTranslations: [],
        censorRatingId: ExampleCensorRatings.G.id,
        runtime: 98,
        displayPriority: 2,
        actorIds: [
            ExamplePeople.RyanGosling.id,
            ExamplePeople.CarrieFisher.id,
            ExamplePeople.JohnBoyega.id
        ],
        directorIds: [
            ExamplePeople.TomHanks.id
        ],
        genreIds: [],
        categories: [FilmCategory.NowShowing, FilmCategory.ComingSoon]
    },
    {
        id: '10',
        title: 'Smallfoot',
        titleTranslations: [],
        plotTranslations: [],
        classificationNoteTranslations: [],
        censorRatingId: ExampleCensorRatings.G.id,
        runtime: 98,
        displayPriority: 2,
        actorIds: [
            ExamplePeople.RyanGosling.id,
            ExamplePeople.CarrieFisher.id,
            ExamplePeople.JohnBoyega.id
        ],
        directorIds: [
            ExamplePeople.TomHanks.id
        ],
        genreIds: [ExampleGenres.Comedy.id],
        categories: [FilmCategory.NowShowing, FilmCategory.ComingSoon]
    },
    {
        id: '11',
        title: 'Bad Times at the El Roayle',
        titleTranslations: [],
        plotTranslations: [],
        classificationNoteTranslations: [],
        censorRatingId: ExampleCensorRatings.M.id,
        runtime: 98,
        displayPriority: 2,
        actorIds: [
            ExamplePeople.RyanGosling.id,
            ExamplePeople.CarrieFisher.id,
            ExamplePeople.JohnBoyega.id
        ],
        directorIds: [
            ExamplePeople.TomHanks.id
        ],
        genreIds: [ExampleGenres.Action.id],
        categories: [FilmCategory.NowShowing, FilmCategory.ComingSoon]
    },
];
export var exampleFilm = exampleFilms[0];
