// @vista/web-widgets/lib/example-data/browsing/showtimes.js
import { parse } from 'date-fns';
import { exampleBusinessDate } from './business-date';
import { exampleFilms } from './film';
import { ExampleShowtimeAttributes } from './showtime-attributes';
import { exampleSites } from './site';
export var exampleShowtimes = [
    {
        id: '1',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T09:15:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.Captions.id,
            ExampleShowtimeAttributes.Atmos.id,
            ExampleShowtimeAttributes.ThreeD.id,
            ExampleShowtimeAttributes.Hfr.id,
        ]
    },
    {
        id: '2',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T11:00:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.Captions.id,
            ExampleShowtimeAttributes.Atmos.id,
            ExampleShowtimeAttributes.Cineplatinum.id,
            ExampleShowtimeAttributes.Saver.id,
            ExampleShowtimeAttributes.ThreeD.id,
            ExampleShowtimeAttributes.Hfr.id,
        ]
    },
    {
        id: '3',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T12:55:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.GoldClass.id,
            ExampleShowtimeAttributes.SuperSaver.id,
        ]
    },
    {
        id: '4',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T15:15:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.Captions.id,
            ExampleShowtimeAttributes.Atmos.id,
            ExampleShowtimeAttributes.Saver.id,
        ]
    },
    {
        id: '5',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[1].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T16:30:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.Captions.id,
            ExampleShowtimeAttributes.Atmos.id,
            ExampleShowtimeAttributes.GoldClass.id,
        ]
    },
    {
        id: '6',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[1].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T19:20:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.SuperSaver.id,
            ExampleShowtimeAttributes.Hfr.id,
            ExampleShowtimeAttributes.ThreeD.id,
            ExampleShowtimeAttributes.Captions.id,
        ]
    },
    {
        id: '7',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T19:35:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.ImaxAlt.id,
        ]
    },
    {
        id: '8',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T21:00:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.Atmos.id,
            ExampleShowtimeAttributes.Cineplatinum.id,
        ]
    },
    {
        id: '9',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T21:15:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.Atmos.id,
        ]
    },
    {
        id: '10',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T22:05:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
        ]
    },
    {
        id: '11',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-15T23:00:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
        ]
    },
    {
        id: '12',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-16T00:01:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.ThreeDhero.id,
        ]
    },
    {
        id: '13',
        businessDate: exampleBusinessDate,
        filmId: exampleFilms[0].id,
        siteId: exampleSites[0].id,
        isAllocatedSeating: true,
        screenName: 'Theatre 6',
        screenNameTranslations: [],
        startTime: parse('2019-05-16T02:15:00'),
        attributeIds: [
            ExampleShowtimeAttributes.Hearing.id,
            ExampleShowtimeAttributes.Recliner.id,
            ExampleShowtimeAttributes.Captions.id,
            ExampleShowtimeAttributes.Atmos.id,
        ]
    },
];
