// @vista/web-widgets/lib/example-data/browsing/showtime-attributes.js
export var ExampleShowtimeAttributes;
(function (ExampleShowtimeAttributes) {
    ExampleShowtimeAttributes.Hearing = {
        id: 'HO001',
        shortName: 'Hearing',
        shortNameTranslations: [],
        description: 'Inductive Hearing Loop',
        descriptionTranslations: [],
        displayPriority: 4,
        isPromoted: false
    };
    ExampleShowtimeAttributes.Captions = {
        id: 'HO002',
        shortName: 'CC',
        shortNameTranslations: [],
        description: 'Closed Captioning',
        descriptionTranslations: [],
        displayPriority: 4,
        isPromoted: false
    };
    ExampleShowtimeAttributes.Atmos = {
        id: 'HO003',
        shortName: 'Atmos',
        shortNameTranslations: [],
        description: 'Dolby Atmos Sound',
        descriptionTranslations: [],
        displayPriority: 3,
        isPromoted: false
    };
    ExampleShowtimeAttributes.Cineplatinum = {
        id: 'HO004',
        shortName: 'CP',
        shortNameTranslations: [],
        description: 'Cine-Platinum',
        descriptionTranslations: [],
        displayPriority: 1,
        isPromoted: false
    };
    ExampleShowtimeAttributes.Saver = {
        id: 'HO005',
        shortName: 'Saver',
        shortNameTranslations: [],
        description: 'Saver',
        descriptionTranslations: [],
        displayPriority: 1,
        isPromoted: true
    };
    ExampleShowtimeAttributes.ThreeDhero = {
        id: 'HO006',
        shortName: '3Dii',
        shortNameTranslations: [],
        description: '3D',
        descriptionTranslations: [],
        displayPriority: 1,
        isPromoted: false
    };
    ExampleShowtimeAttributes.ThreeD = {
        id: 'HO007',
        shortName: '3D',
        shortNameTranslations: [],
        description: '3D',
        descriptionTranslations: [],
        displayPriority: 2,
        isPromoted: false
    };
    ExampleShowtimeAttributes.SuperSaver = {
        id: 'HO008',
        shortName: 'SuperSaver',
        shortNameTranslations: [],
        description: 'Super Saver',
        descriptionTranslations: [],
        displayPriority: 2,
        isPromoted: true
    };
    ExampleShowtimeAttributes.Imax = {
        id: 'HO009',
        shortName: 'IMAX',
        shortNameTranslations: [],
        description: 'IMAX 70mm',
        descriptionTranslations: [],
        displayPriority: 1,
        isPromoted: false
    };
    ExampleShowtimeAttributes.ImaxAlt = {
        id: 'HO010',
        shortName: 'IMAXDIGITAL',
        shortNameTranslations: [],
        description: 'IMAX with Laser',
        descriptionTranslations: [],
        displayPriority: 1,
        isPromoted: false
    };
    ExampleShowtimeAttributes.GoldClass = {
        id: 'HO011',
        shortName: 'GC',
        shortNameTranslations: [],
        description: 'Gold Class',
        descriptionTranslations: [],
        displayPriority: 1,
        isPromoted: false
    };
    ExampleShowtimeAttributes.Hfr = {
        id: 'HO012',
        shortName: 'HFR',
        shortNameTranslations: [],
        description: 'High Frame Rate',
        descriptionTranslations: [],
        displayPriority: 2,
        isPromoted: false
    };
    ExampleShowtimeAttributes.Recliner = {
        id: 'HO013',
        shortName: 'RECLINER',
        shortNameTranslations: [],
        description: 'Recliner Seats',
        descriptionTranslations: [],
        displayPriority: 3,
        isPromoted: false
    };
})(ExampleShowtimeAttributes || (ExampleShowtimeAttributes = {}));
export var exampleShowtimeAttributes = [
    ExampleShowtimeAttributes.Hearing,
    ExampleShowtimeAttributes.Captions,
    ExampleShowtimeAttributes.Atmos,
    ExampleShowtimeAttributes.Cineplatinum,
    ExampleShowtimeAttributes.Saver,
    ExampleShowtimeAttributes.ThreeDhero,
    ExampleShowtimeAttributes.ThreeD,
    ExampleShowtimeAttributes.SuperSaver,
    ExampleShowtimeAttributes.Imax,
    ExampleShowtimeAttributes.ImaxAlt,
    ExampleShowtimeAttributes.GoldClass,
    ExampleShowtimeAttributes.Hfr,
    ExampleShowtimeAttributes.Recliner
];
