// @vista/web-widgets/lib/example-data/browsing/site.js
export var exampleSites = [
    {
        id: '1',
        name: 'Queen Street',
        nameTranslations: [],
        address: {
            line1: '80 Queen St',
            city: 'Auckland Central'
        },
        contactDetails: {
            phoneNumber1: '+64 9 653 7443'
        }
    },
    {
        id: '2',
        name: 'Albany',
        nameTranslations: [],
        address: {
            line1: '14 Don McKinnon Dr',
            city: 'Albany'
        },
        contactDetails: {
            phoneNumber1: '+64 9 982 8221'
        }
    },
    {
        id: '3',
        name: 'Takapuna',
        nameTranslations: [],
        address: {
            line1: '52 Anzac St',
            city: 'Takapuna'
        },
        contactDetails: {
            phoneNumber1: '+64 9 332 2131'
        }
    },
    {
        id: '4',
        name: 'Wairau',
        nameTranslations: [],
        address: {
            line1: '24 Link Dr',
            city: 'Wairau Valley'
        },
        contactDetails: {
            phoneNumber1: '+64 9 463 0922'
        }
    },
    {
        id: '5',
        name: 'Silverdale',
        nameTranslations: [],
        address: {
            line1: '75 Silverdale St',
            city: 'Silverdale'
        },
        contactDetails: {
            phoneNumber1: '+64 9 123 4332'
        }
    },
    {
        id: '6',
        name: 'Milford',
        nameTranslations: [],
        address: {
            line1: '24 Milford Rd',
            city: 'Milford'
        },
        contactDetails: {
            phoneNumber1: '+64 9 938 2123'
        }
    },
    {
        id: '7',
        name: 'Panmure',
        nameTranslations: [],
        address: {
            line1: '16 Clifton Rd',
            city: 'Panmure'
        },
        contactDetails: {
            phoneNumber1: '+64 9 634 3423'
        }
    },
    {
        id: '8',
        name: 'New Lynn',
        nameTranslations: [],
        address: {
            line1: '3058 Great North Rd',
            city: 'New Lynn'
        },
        contactDetails: {
            phoneNumber1: '+64 9 442 4365'
        }
    },
    {
        id: '9',
        name: 'Henderson',
        nameTranslations: [],
        address: {
            line1: '7 Catherine St',
            city: 'Henderson'
        },
        contactDetails: {
            phoneNumber1: '+64 9 523 4231'
        }
    },
    {
        id: '10',
        name: 'Sylvia Park',
        nameTranslations: [],
        address: {
            line1: '393 Mount Wellington Hwy',
            city: 'Mount Wellington'
        },
        contactDetails: {
            phoneNumber1: '+64 9 534 221'
        }
    },
];
export var exampleSite = exampleSites[0];
