// @vista/web-widgets/lib/api/src/mappers/loyalty/map-loyalty-transactions-to-member-transactions.js
var mapDateStringToDate = function (dateString) { return dateString ? new Date(dateString) : undefined; };
var mapMxLoyaltyMemberTransactionItem = function (item) { return ({
    transactionLineNumber: item.transactionLineNumber,
    itemId: item.itemId,
    price: item.price,
    quantity: item.quantity
}); };
var mapMxLoyaltyMemberTransactionTicket = function (ticket) { return ({
    transactionLineNumber: ticket.transactionLineNumber,
    ticketTypeId: ticket.ticketTypeId,
    price: ticket.price,
    quantity: ticket.quantity
}); };
var mapMxLoyaltyMemberTransactionShowtime = function (showtime) { return ({
    showtimeId: showtime.showtimeId,
    tickets: showtime.tickets.map(function (x) { return mapMxLoyaltyMemberTransactionTicket(x); })
}); };
var mapMxLoyaltyMemberTransactionPointsAdjustment = function (pointsAdjustment) { return ({
    balanceTypeId: pointsAdjustment.balanceTypeId,
    adjustmentValue: pointsAdjustment.adjustmentValue,
    balance: pointsAdjustment.balance
}); };
var mapMxLoyaltyMemberTransactionsToMemberTransaction = function (transaction) { return ({
    id: transaction.id,
    bookingId: transaction.bookingId,
    transactionType: transaction.transactionType,
    transactionTime: mapDateStringToDate(transaction.transactionTime),
    showtimes: transaction.showtimes.map(function (x) { return mapMxLoyaltyMemberTransactionShowtime(x); }),
    items: transaction.items.map(function (x) { return mapMxLoyaltyMemberTransactionItem(x); }),
    pointsAdjustments: transaction.pointsAdjustments.map(function (x) { return mapMxLoyaltyMemberTransactionPointsAdjustment(x); })
}); };
export var mapMxLoyaltyMemberTransactionsToMemberTransactions = function (transactions) { return (transactions.map(function (x) { return mapMxLoyaltyMemberTransactionsToMemberTransaction(x); })); };
