// @vista/web-widgets/lib/api/src/mappers/loyalty/map-loyalty-transactions-related-data.js
import { arrayToMap } from '../../utils/array-utils';
export var mapMxLoyaltyRelatedDataToLoyaltyRelatedData = function (relatedData) {
    return {
        sites: arrayToMap(relatedData.sites.map(function (x) { return mapMxLoyaltySite(x); })),
        films: arrayToMap(relatedData.films.map(function (x) { return mapMxLoyaltyFilm(x); })),
        showtimes: arrayToMap(relatedData.showtimes.map(function (x) { return mapMxLoyaltyShowtime(x); })),
        ticketTypes: arrayToMap(relatedData.ticketTypes.map(function (x) { return mapMxLoyaltyTicketType(x); })),
        items: arrayToMap(relatedData.items.map(function (x) { return mapMxLoyaltyItemModel(x); })),
        balanceTypes: arrayToMap(relatedData.balanceTypes.map(function (x) { return mapMxLoyaltyBalanceType(x); })),
    };
};
var mapMxLoyaltySite = function (item) { return ({
    id: item.id,
    name: item.name
}); };
var mapMxLoyaltyFilm = function (item) { return ({
    id: item.id,
    title: item.title
}); };
var mapMxLoyaltyShowtime = function (item) { return ({
    id: item.id,
    filmId: item.filmId,
    siteId: item.siteId
}); };
var mapMxLoyaltyTicketType = function (item) { return ({
    id: item.id,
    description: item.description
}); };
var mapMxLoyaltyItemModel = function (item) { return ({
    id: item.id,
    description: item.description
}); };
var mapMxLoyaltyBalanceType = function (item) { return ({
    id: item.id,
    name: item.name
}); };
