// @vista/web-widgets/lib/api/src/mappers/loyalty/map-member-detail-to-loyalty-detail.js
var mapDateStringToDate = function (dateString) {
    if (dateString) {
        var dateNum = Date.parse(dateString);
        if (!isNaN(dateNum))
            return new Date(dateNum);
    }
    return undefined;
};
var mapDefaultMemberBalance = function (memberBalance) {
    if (memberBalance) {
        var balance = memberBalance.find(function (mb) { return mb.isDefault; });
        if (balance)
            return balance;
    }
    return undefined;
};
var mapExpiryPointsByBalanceTypeId = function (expiryPoints, defaultMemberBalance) {
    if (expiryPoints && defaultMemberBalance) {
        var expPoints = expiryPoints.filter(function (ep) { return ep.balanceTypeId === defaultMemberBalance.balanceTypeId; });
        var closestExpiryPoints = (expPoints && expPoints.length > 0) && expPoints.reduce(function (a, b) { return a.expireOn < b.expireOn ? a : b; });
        if (closestExpiryPoints)
            return closestExpiryPoints;
    }
    return undefined;
};
export var mapMxLoyaltyMemberToLoyaltyMember = function (member) {
    var defaultMemberBalance = mapDefaultMemberBalance(member.memberBalance);
    return {
        memberId: member.memberId,
        givenName: member.firstName,
        familyName: member.lastName,
        fullName: member.fullName,
        email: member.email,
        mobilePhone: member.mobilePhone,
        homePhone: member.homePhone,
        dateOfBirth: mapDateStringToDate(member.dateOfBirth),
        memberLevelName: member.memberLevelName,
        cardNumber: member.cardNumber,
        clubName: member.clubName,
        memberBalance: defaultMemberBalance,
        expiryPoints: mapExpiryPointsByBalanceTypeId(member.expiryPoints, defaultMemberBalance)
    };
};
