// @vista/web-widgets/lib/example-data/loyalty/member-transactions/loyalty-member-service.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { exampleRelatedData, exampleRelatedDataPage2, exampleRelatedDataPage3, exampleTransactionHistory, exampleTransactionHistoryPage2, exampleTransactionHistoryPage3 } from './member-transaction';
export var exampleLoyaltyMemberService = {
    getSignedInMemberDetails: function () { return Promise.resolve(undefined); },
    loadTransactionHistory: function (_numberOfTransactionsToLoad, top) {
        var transactionHistoryResponse = {
            transactions: exampleTransactionHistory,
            relatedData: exampleRelatedData,
            isFullyLoaded: false
        };
        if (top === exampleTransactionHistory.length) {
            transactionHistoryResponse = __assign({}, transactionHistoryResponse, { transactions: exampleTransactionHistoryPage2, relatedData: exampleRelatedDataPage2 });
        }
        if (top === (exampleTransactionHistory.length + exampleTransactionHistoryPage2.length)) {
            transactionHistoryResponse = {
                transactions: exampleTransactionHistoryPage3,
                relatedData: exampleRelatedDataPage3,
                isFullyLoaded: true
            };
        }
        return Promise.resolve(transactionHistoryResponse);
    }
};
