// @vista/web-widgets/lib/example-data/loyalty/member-transactions/member-transaction.js
export var exampleTransactionHistory = [
    {
        id: 33330,
        bookingId: 'WSHDMSG',
        transactionType: 'Purchase',
        transactionTime: new Date('2019-01-23T17:59:36'),
        showtimes: [
            {
                showtimeId: '1-s1',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 1
                    },
                    {
                        transactionLineNumber: 2,
                        ticketTypeId: 't2',
                        price: 12,
                        quantity: 1
                    }
                ]
            }
        ],
        items: [
            {
                transactionLineNumber: 3,
                itemId: 'i1',
                price: 3.8,
                quantity: 1
            },
            {
                transactionLineNumber: 4,
                itemId: 'i2',
                price: 3.8,
                quantity: 1
            },
            {
                transactionLineNumber: 5,
                itemId: 'i3',
                price: 1,
                quantity: 1
            },
            {
                transactionLineNumber: 6,
                itemId: 'i2',
                price: 1,
                quantity: 1
            }
        ],
        pointsAdjustments: [
            {
                balanceTypeId: '1',
                adjustmentValue: 150,
                balance: 1000
            },
            {
                balanceTypeId: '2',
                adjustmentValue: 45,
                balance: 1000
            }
        ]
    },
    {
        id: 33329,
        bookingId: 'WXJ8H6L',
        transactionType: 'Purchase',
        transactionTime: new Date('2019-01-22T17:44:25'),
        showtimes: [
            {
                showtimeId: '2-s1',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 2
                    }
                ]
            }
        ],
        items: [],
        pointsAdjustments: [
            {
                balanceTypeId: '2',
                adjustmentValue: 80,
                balance: 15971.79
            }
        ]
    },
    {
        id: 33328,
        bookingId: 'WQ7WWQ6',
        transactionType: 'Purchase',
        transactionTime: new Date('2019-01-21T17:36:48'),
        showtimes: [
            {
                showtimeId: '3-s1',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 2
                    },
                ]
            }
        ],
        items: [
            {
                transactionLineNumber: 3,
                itemId: 'i1',
                price: 3.8,
                quantity: 1
            },
        ],
        pointsAdjustments: [
            {
                balanceTypeId: '2',
                adjustmentValue: -200,
                balance: 15891.79
            }
        ]
    },
];
export var exampleRelatedData = {
    sites: new Map([
        ['s1', { id: 's1', name: { text: 'ABC Cinemas St Louis' } }],
        ['s2', { id: 's2', name: { text: 'ABC Wairau' } }],
    ]),
    films: new Map([
        ['f1', { id: 'f1', title: { text: 'Star Wars: The Last Jedi' } }],
        ['f2', { id: 'f2', title: { text: 'Goosebump 2: Haunted Halloween' } }],
        ['f3', { id: 'f3', title: { text: 'Johnny English Strikes Again' } }],
        ['f4', { id: 'f4', title: { text: 'First Man' } }],
    ]),
    showtimes: new Map([
        ['1-s1', { id: '1-s1', siteId: 's1', filmId: 'f1' }],
        ['2-s1', { id: '2-s1', siteId: 's2', filmId: 'f2' }],
        ['3-s1', { id: '3-s1', siteId: 's1', filmId: 'f1' }],
    ]),
    ticketTypes: new Map([
        ['t1', { id: 't1', description: { text: 'Adult ticket' } }],
        ['t2', { id: 't2', description: { text: 'Child ticket' } }],
    ]),
    items: new Map([
        ['i1', { id: 'i1', description: { text: 'Popcorn' } }],
        ['i2', { id: 'i2', description: { text: 'Coca-Cola' } }],
        ['i3', { id: 'i3', description: { text: 'Fanta' } }],
    ]),
    balanceTypes: new Map([
        ['1', { id: '1', name: { text: 'Balance1' } }],
        ['2', { id: '2', name: { text: 'Balance2' } }],
    ]),
};
export var exampleTransactionHistoryPage2 = [
    {
        id: 302,
        bookingId: 'WSHDMSG',
        transactionType: 'Purchase',
        transactionTime: new Date('2019-01-15T17:59:36'),
        showtimes: [
            {
                showtimeId: '4-s3',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 2
                    }
                ]
            }
        ],
        items: [],
        pointsAdjustments: [
            {
                balanceTypeId: '1',
                adjustmentValue: 350,
                balance: 1000
            }
        ]
    },
    {
        id: 301,
        bookingId: 'WXJ8H6L',
        transactionType: 'Purchase',
        transactionTime: new Date('2019-01-14T17:44:25'),
        showtimes: [
            {
                showtimeId: '5-s1',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 2
                    }
                ]
            }
        ],
        items: [],
        pointsAdjustments: [
            {
                balanceTypeId: '2',
                adjustmentValue: -350,
                balance: 15971.79
            }
        ]
    },
    {
        id: 300,
        bookingId: 'WQ7WWQ6',
        transactionType: 'Purchase',
        transactionTime: new Date('2019-01-13T17:36:48'),
        showtimes: [
            {
                showtimeId: '6-s1',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 2
                    },
                ]
            }
        ],
        items: [
            {
                transactionLineNumber: 3,
                itemId: 'i1',
                price: 3.8,
                quantity: 1
            },
        ],
        pointsAdjustments: [
            {
                balanceTypeId: '2',
                adjustmentValue: -200,
                balance: 15891.79
            }
        ]
    },
];
export var exampleRelatedDataPage2 = {
    sites: new Map([
        ['s3', { id: 's3', name: { text: 'ABC Queen Street' } }],
    ]),
    films: new Map([
        ['f4', { id: 'f4', title: { text: 'First Man' } }],
    ]),
    showtimes: new Map([
        ['4-s3', { id: '4-s3', siteId: 's3', filmId: 'f1' }],
        ['5-s1', { id: '5-s1', siteId: 's1', filmId: 'f4' }],
        ['6-s1', { id: '6-s1', siteId: 's1', filmId: 'f4' }],
    ]),
    ticketTypes: new Map(),
    items: new Map([
        ['i1', { id: 'i1', description: { text: 'Popcorn' } }],
        ['i2', { id: 'i2', description: { text: 'Coca-Cola' } }],
        ['i3', { id: 'i3', description: { text: 'Fanta' } }],
    ]),
    balanceTypes: new Map([
        ['1', { id: '1', name: { text: 'Balance1' } }],
        ['2', { id: '2', name: { text: 'Balance2' } }],
    ]),
};
export var exampleTransactionHistoryPage3 = [
    {
        id: 202,
        bookingId: 'WSHDMSG',
        transactionType: 'Purchase',
        transactionTime: new Date('2019-01-03T17:59:36'),
        showtimes: [
            {
                showtimeId: '7-s3',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 2
                    }
                ]
            }
        ],
        items: [],
        pointsAdjustments: [
            {
                balanceTypeId: '1',
                adjustmentValue: 350,
                balance: 1000
            }
        ]
    },
    {
        id: 201,
        bookingId: 'WXJ8H6L',
        transactionType: 'Purchase',
        transactionTime: new Date('2019-01-02T17:44:25'),
        showtimes: [
            {
                showtimeId: '8-s1',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't2',
                        price: 2,
                        quantity: 2
                    }
                ]
            }
        ],
        items: [],
        pointsAdjustments: [
            {
                balanceTypeId: '2',
                adjustmentValue: -350,
                balance: 15971.79
            }
        ]
    },
    {
        id: 200,
        bookingId: 'WQ7WWQ6',
        transactionType: 'Purchase',
        transactionTime: new Date('2019-01-01T17:36:48'),
        showtimes: [
            {
                showtimeId: '9-s1',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 2
                    },
                ]
            }
        ],
        items: [
            {
                transactionLineNumber: 3,
                itemId: 'i1',
                price: 3.8,
                quantity: 1
            },
        ],
        pointsAdjustments: [
            {
                balanceTypeId: '2',
                adjustmentValue: -200,
                balance: 15891.79
            }
        ]
    },
];
export var exampleRelatedDataPage3 = {
    sites: new Map([
        ['s1', { id: 's1', name: { text: 'ABC Cinemas St Louis' } }],
        ['s3', { id: 's3', name: { text: 'ABC Queen Street' } }],
    ]),
    films: new Map([
        ['f4', { id: 'f4', title: { text: 'First Man' } }],
    ]),
    showtimes: new Map([
        ['7-s3', { id: '7-s3', siteId: 's3', filmId: 'f1' }],
        ['8-s1', { id: '8-s1', siteId: 's1', filmId: 'f4' }],
        ['9-s1', { id: '9-s1', siteId: 's1', filmId: 'f4' }],
    ]),
    ticketTypes: new Map(),
    items: new Map(),
    balanceTypes: new Map([
        ['1', { id: '1', name: { text: 'Balance1' } }],
        ['2', { id: '2', name: { text: 'Balance2' } }],
    ]),
};
export var exampleMergedRelatedData = {
    sites: new Map([
        ['s1', { id: 's1', name: { text: 'ABC Cinemas St Louis' } }],
        ['s3', { id: 's3', name: { text: 'ABC Queen Street' } }],
    ]),
    films: new Map([
        ['f4', { id: 'f4', title: { text: 'First Man' } }],
    ]),
    showtimes: new Map([
        ['4-s3', { id: '4-s3', siteId: 's3', filmId: 'f1' }],
        ['5-s1', { id: '5-s1', siteId: 's1', filmId: 'f4' }],
        ['6-s1', { id: '6-s1', siteId: 's1', filmId: 'f4' }],
        ['7-s3', { id: '7-s3', siteId: 's3', filmId: 'f1' }],
        ['8-s1', { id: '8-s1', siteId: 's1', filmId: 'f4' }],
        ['9-s1', { id: '9-s1', siteId: 's1', filmId: 'f4' }],
    ]),
    ticketTypes: new Map(),
    items: new Map([
        ['i1', { id: 'i1', description: { text: 'Popcorn' } }],
        ['i2', { id: 'i2', description: { text: 'Coca-Cola' } }],
        ['i3', { id: 'i3', description: { text: 'Fanta' } }],
    ]),
    balanceTypes: new Map([
        ['1', { id: '1', name: { text: 'Balance1' } }],
        ['2', { id: '2', name: { text: 'Balance2' } }],
    ]),
};
