// @vista/web-widgets/lib/example-data/loyalty/member-transactions/mx-member-transaction.js
export var exampleMxMemberTransaction = [{
        id: 33330,
        bookingId: 'WSHDMSG',
        transactionType: 'Purchase',
        transactionTime: '2019-01-23T17:59:36',
        showtimes: [
            {
                showtimeId: '1-s1',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 1
                    },
                    {
                        transactionLineNumber: 2,
                        ticketTypeId: 't2',
                        price: 12,
                        quantity: 1
                    }
                ]
            }
        ],
        items: [
            {
                transactionLineNumber: 3,
                itemId: 'i1',
                price: 3.8,
                quantity: 1
            },
            {
                transactionLineNumber: 4,
                itemId: 'i2',
                price: 3.8,
                quantity: 1
            },
            {
                transactionLineNumber: 5,
                itemId: 'i3',
                price: 1,
                quantity: 1
            },
            {
                transactionLineNumber: 6,
                itemId: 'i2',
                price: 1,
                quantity: 1
            }
        ],
        pointsAdjustments: [
            {
                balanceTypeId: '1',
                adjustmentValue: 150,
                balance: 1000
            },
            {
                balanceTypeId: '2',
                adjustmentValue: 45,
                balance: 1000
            }
        ]
    },
    {
        id: 33329,
        bookingId: 'WXJ8H6L',
        transactionType: 'Purchase',
        transactionTime: '2019-01-22T17:44:25',
        showtimes: [
            {
                showtimeId: '2-s1',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 2
                    }
                ]
            }
        ],
        items: [],
        pointsAdjustments: [
            {
                balanceTypeId: '2',
                adjustmentValue: 80,
                balance: 15971.79
            }
        ]
    },
    {
        id: 33328,
        bookingId: 'WQ7WWQ6',
        transactionType: 'Purchase',
        transactionTime: '2019-01-21T17:36:48',
        showtimes: [
            {
                showtimeId: '3-s1',
                tickets: [
                    {
                        transactionLineNumber: 1,
                        ticketTypeId: 't1',
                        price: 2,
                        quantity: 2
                    },
                ]
            }
        ],
        items: [
            {
                transactionLineNumber: 3,
                itemId: 'i1',
                price: 3.8,
                quantity: 1
            },
        ],
        pointsAdjustments: [
            {
                balanceTypeId: '2',
                adjustmentValue: -200,
                balance: 15891.79
            }
        ]
    }];
export var exampleMxRelatedData = {
    sites: [
        { id: 's1', name: { text: 'ABC Cinemas St Louis' } },
        { id: 's2', name: { text: 'ABC Wairau' } },
    ],
    films: [
        { id: 'f1', title: { text: 'Star Wars: The Last Jedi' } },
        { id: 'f2', title: { text: 'Goosebump 2: Haunted Halloween' } },
        { id: 'f3', title: { text: 'Johnny English Strikes Again' } },
        { id: 'f4', title: { text: 'First Man' } },
    ],
    showtimes: [
        { id: '1-s1', siteId: 's1', filmId: 'f1' },
        { id: '2-s1', siteId: 's2', filmId: 'f2' },
        { id: '3-s1', siteId: 's1', filmId: 'f1' },
    ],
    ticketTypes: [
        { id: 't1', description: { text: 'Adult ticket' } },
        { id: 't2', description: { text: 'Child ticket' } },
    ],
    items: [
        { id: 'i1', description: { text: 'Popcorn' } },
        { id: 'i2', description: { text: 'Coca-Cola' } },
        { id: 'i3', description: { text: 'Fanta' } },
    ],
    balanceTypes: [
        { id: '1', name: { text: 'Balance1' } },
        { id: '2', name: { text: 'Balance2' } },
    ],
};
