// @vista/web-widgets/lib/widgets/src/components/film-list/components/film-list-carousel.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { bemClassBuilderFactory } from '../../../utils/bem-class-builder';
import { Carousel } from '../../carousel/carousel';
import { FilmListFilm } from './film-list-film';
var FilmListCarousel = /** @class */ (function (_super) {
    __extends(FilmListCarousel, _super);
    function FilmListCarousel() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FilmListCarousel.prototype.render = function () {
        var _a = this.props, films = _a.films, onFilmClicked = _a.onFilmClicked;
        var b = bemClassBuilderFactory('v-film-list-carousel');
        var carouselProps = {
            showDots: false,
            refreshOnUpdate: true,
            settings: {
                slidesToShow: 2.2,
                slidesToScroll: 'auto',
                responsive: [
                    {
                        breakpoint: 550,
                        settings: {
                            slidesToShow: 3.2
                        }
                    },
                    {
                        breakpoint: 767,
                        settings: {
                            slidesToShow: 4.2
                        }
                    },
                    {
                        breakpoint: 950,
                        settings: {
                            slidesToShow: 5.2
                        }
                    },
                    {
                        breakpoint: 1200,
                        settings: {
                            slidesToShow: 6.2
                        }
                    }
                ]
            }
        };
        return (React.createElement("div", { className: b.block() },
            React.createElement(Carousel, __assign({}, carouselProps), films.map(function (film) {
                return React.createElement(FilmListFilm, { key: film.id, film: film, onClick: onFilmClicked });
            }))));
    };
    return FilmListCarousel;
}(React.Component));
export { FilmListCarousel };
