// @vista/web-widgets/lib/example-data/seat-first-order.js
// TODO: remove any's by using an order-fixture builder.
var orderWithSeatsInDifferentAreaCategories = {
    bookingFee: 0,
    showtimes: [
        {
            showtimeId: '0000001',
            filmId: '1',
            screenName: 'Screen 1',
            date: new Date(2017, 10, 24),
            seats: [
                { rowName: 'A', seatName: '1', areaCategoryCode: 'AreaCategory1' },
                { rowName: 'A', seatName: '2', areaCategoryCode: 'AreaCategory2' },
                { rowName: 'A', seatName: '3', areaCategoryCode: 'AreaCategory3' }
            ],
            tickets: [
                { ticketTypeId: '0001', areaCategoryCode: 'AreaCategory1', description: 'Adult', price: 15.00, quantity: 0 },
                { ticketTypeId: '0002', areaCategoryCode: 'AreaCategory2', description: 'Child', price: 11.00, quantity: 0 },
                { ticketTypeId: '0003', areaCategoryCode: 'AreaCategory3', description: 'Senior Citizen', price: 12.50, quantity: 0 }
            ]
        }
    ]
};
var orderWithSeatsInOneAreaCategory = {
    bookingFee: 0,
    showtimes: [
        {
            showtimeId: '0000001',
            filmId: '1',
            screenName: 'Screen 1',
            date: new Date(2017, 10, 24),
            seats: [
                { rowName: 'A', seatName: '1', areaCategoryCode: 'AreaCategory1' },
                { rowName: 'A', seatName: '2', areaCategoryCode: 'AreaCategory1' },
                { rowName: 'A', seatName: '3', areaCategoryCode: 'AreaCategory1' }
            ],
            tickets: [
                { ticketTypeId: '0001', areaCategoryCode: 'AreaCategory1', description: 'Adult', price: 15.00, quantity: 0, },
                { ticketTypeId: '0002', areaCategoryCode: 'AreaCategory2', description: 'Child', price: 11.00, quantity: 0, },
                { ticketTypeId: '0003', areaCategoryCode: 'AreaCategory3', description: 'Senior Citizen', price: 12.50, quantity: 0, }
            ]
        }
    ]
};
var noSeatsAssignedOrder = {
    bookingFee: 0,
    showtimes: [
        {
            showtimeId: '0000001',
            filmId: '1',
            screenName: 'Screen 1',
            date: new Date(2017, 10, 24),
            tickets: [
                { ticketTypeId: '0001', areaCategoryCode: 'AreaCategory1', description: 'Adult', price: 15.00, quantity: 0, },
                { ticketTypeId: '0002', areaCategoryCode: 'AreaCategory2', description: 'Child', price: 11.00, quantity: 0, },
                { ticketTypeId: '0003', areaCategoryCode: 'AreaCategory3', description: 'Senior Citizen', price: 12.50, quantity: 0, }
            ]
        }
    ]
};
var moreTicketsAssignedThanSeats = {
    bookingFee: 0,
    showtimes: [
        {
            showtimeId: '0000001',
            filmId: '1',
            screenName: 'Screen 1',
            date: new Date(2017, 10, 24),
            seats: [
                { rowName: 'A', seatName: '1', areaCategoryCode: 'AreaCategory1' },
                { rowName: 'A', seatName: '2', areaCategoryCode: 'AreaCategory1' },
                { rowName: 'A', seatName: '3', areaCategoryCode: 'AreaCategory1' }
            ],
            tickets: [
                { ticketTypeId: '0001', areaCategoryCode: 'AreaCategory1', description: 'Adult', price: 15.00, quantity: 2, },
                { ticketTypeId: '0002', areaCategoryCode: 'AreaCategory1', description: 'Child', price: 11.00, quantity: 2, }
            ]
        }
    ]
};
var oneAreaCategoryWithPartiallyAssignedTickets = {
    bookingFee: 0,
    showtimes: [
        {
            showtimeId: '0000001',
            filmId: '1',
            screenName: 'Screen 1',
            date: new Date(2017, 10, 24),
            seats: [
                { rowName: 'A', seatName: '1', areaCategoryCode: 'AreaCategory1' },
                { rowName: 'A', seatName: '2', areaCategoryCode: 'AreaCategory1' },
                { rowName: 'A', seatName: '3', areaCategoryCode: 'AreaCategory1' }
            ],
            tickets: [
                { ticketTypeId: '0001', areaCategoryCode: 'AreaCategory1', description: 'Adult', price: 15.00, quantity: 1, },
                { ticketTypeId: '0002', areaCategoryCode: 'AreaCategory1', description: 'Child', price: 11.00, quantity: 1, }
            ]
        }
    ]
};
export { orderWithSeatsInDifferentAreaCategories, orderWithSeatsInOneAreaCategory, noSeatsAssignedOrder, oneAreaCategoryWithPartiallyAssignedTickets, moreTicketsAssignedThanSeats };
