// @vista/web-widgets/lib/example-data/vouchers.js
import { VoucherStatus } from '../types/vouchers/voucher-status';
import { exampleVoucherTicketType1, exampleVoucherTicketType2 } from './order/order-ticket-type';
export var exampleVoucherSingleUse = {
    barcode: 'AD0010000001',
    name: 'Adult voucher',
    ticketRedemptions: {
        remainingRedemptions: 1,
        ticketTypes: [
            { ticketTypeId: exampleVoucherTicketType1.id, ticketTypeKey: exampleVoucherTicketType1.key }
        ]
    },
    status: VoucherStatus.Valid
};
export var exampleVoucherMultiUse = {
    barcode: 'AD0010000002',
    name: 'Adult voucher',
    ticketRedemptions: {
        remainingRedemptions: 2,
        ticketTypes: [
            { ticketTypeId: exampleVoucherTicketType1.id, ticketTypeKey: exampleVoucherTicketType1.key },
            { ticketTypeId: exampleVoucherTicketType2.id, ticketTypeKey: exampleVoucherTicketType2.key },
        ]
    },
    status: VoucherStatus.Valid
};
export var exampleVoucherExpired = {
    barcode: 'AD0010000003',
    name: 'Adult voucher',
    ticketRedemptions: {
        remainingRedemptions: 0,
        ticketTypes: [
            { ticketTypeId: exampleVoucherTicketType1.id, ticketTypeKey: exampleVoucherTicketType1.key }
        ]
    },
    status: VoucherStatus.Expired
};
export var exampleVoucherInvalid = {
    barcode: 'AD0010000004',
    name: 'Adult voucher',
    ticketRedemptions: {
        remainingRedemptions: 0,
        ticketTypes: [
            { ticketTypeId: exampleVoucherTicketType1.id, ticketTypeKey: exampleVoucherTicketType1.key }
        ]
    },
    status: VoucherStatus.Invalid
};
export var exampleVoucherRedeemed = {
    barcode: 'AD0010000005',
    name: 'Adult voucher',
    ticketRedemptions: {
        remainingRedemptions: 0,
        ticketTypes: [
            { ticketTypeId: exampleVoucherTicketType1.id, ticketTypeKey: exampleVoucherTicketType1.key }
        ]
    },
    status: VoucherStatus.Redeemed
};
export var exampleVoucherWithNoTicketTypes = {
    barcode: 'AD0010000006',
    name: 'Non-ticket voucher',
    ticketRedemptions: {
        remainingRedemptions: 0,
        ticketTypes: []
    },
    status: VoucherStatus.Valid
};
export var exampleVouchers = [
    exampleVoucherSingleUse,
    exampleVoucherMultiUse,
    exampleVoucherExpired,
    exampleVoucherInvalid,
    exampleVoucherRedeemed
];
