// @vista/web-widgets/lib/widgets/src/components/order-cart/bootstrap.js
import { orderStoreInstance } from '../../../../api/src/stores/order/order-store';
import { salesDataStoreInstance } from '../../../../api/src/stores/sales-data/sales-data-store';
import { addWidgetBootstrap } from '../../bootstrap/add-widget-bootstrap';
import { createComponent } from '../../bootstrap/create-component';
import { mapOrderCartAssets } from './mappers/assets-mapper';
import { mapOrderCartConfig } from './mappers/config-mapper';
import { mapOrderCartDisplayText } from './mappers/display-text-mapper';
import { OrderCart } from './order-cart';
/**
 * Renders the Order Cart widget.
 * @param bootstrap
 */
export var orderCart = function (bootstrap) {
    var selector = bootstrap.selector, displayText = bootstrap.displayText, config = bootstrap.config, events = bootstrap.events, assets = bootstrap.assets;
    var props = {
        orderStore: orderStoreInstance,
        salesDataStore: salesDataStoreInstance,
        displayText: mapOrderCartDisplayText(displayText),
        config: mapOrderCartConfig(config),
        assets: mapOrderCartAssets(assets),
        onActionButtonClicked: events && events.onActionButtonClicked
    };
    return createComponent(selector, OrderCart, props);
};
addWidgetBootstrap('orderCart', orderCart);
