// @vista/web-widgets/lib/widgets/src/components/item-picker/example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { orderStoreInstance } from '../../../../api/src/stores/order/order-store';
import '../../scss/themes/light/item-picker.scss';
// import '../../scss/themes/dark/item-picker.scss';
import { ItemPickerMessage, ItemPickerMessageType } from './components/item-picker-message';
import { ItemPickerSkeleton } from './components/item-picker-skeleton';
import { exampleSalesDataStore } from './example/example-sales-data-store';
import { ItemPicker } from './item-picker';
import { defaultItemPickerAssets } from './types/assets';
import { defaultItemPickerDisplayText } from './types/display-text';
var props = {
    siteId: '',
    orderStore: orderStoreInstance,
    salesDataStore: exampleSalesDataStore,
    displayText: defaultItemPickerDisplayText,
    assets: defaultItemPickerAssets,
    onItemClicked: function (item) { console.log("Item clicked: " + item.id); }
};
var testItemPickerMessageProps = {
    messageType: ItemPickerMessageType.FailedToLoadData,
    displayText: defaultItemPickerDisplayText,
    assets: defaultItemPickerAssets
};
var ItemPickerExample = /** @class */ (function (_super) {
    __extends(ItemPickerExample, _super);
    function ItemPickerExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ItemPickerExample.prototype.render = function () {
        return (React.createElement("div", { style: { margin: '0 auto', maxWidth: '1200px', padding: '0 16px' } },
            React.createElement("h1", { className: 'v-example-header' }, "Item picker"),
            React.createElement(ItemPicker, __assign({}, props)),
            React.createElement("br", null),
            React.createElement("h2", { className: 'v-example-header' }, "Skeleton"),
            React.createElement(ItemPickerSkeleton, null),
            React.createElement("br", null),
            React.createElement("h2", { className: 'v-example-header' }, "Message"),
            React.createElement(ItemPickerMessage, __assign({}, testItemPickerMessageProps))));
    };
    return ItemPickerExample;
}(React.Component));
export { ItemPickerExample };
