// @vista/web-widgets/lib/widgets/src/components/site-details/components/site-details-message.js
import * as React from 'react';
import { Message } from '../../message/message';
export var SiteDetailsMessageType;
(function (SiteDetailsMessageType) {
    SiteDetailsMessageType["FailedToLoadData"] = "FailedToLoadData";
})(SiteDetailsMessageType || (SiteDetailsMessageType = {}));
function getIconName(messageType, assets) {
    switch (messageType) {
        case SiteDetailsMessageType.FailedToLoadData:
            return assets.failedToLoadDataIconName;
    }
}
function getTitleString(messageType, displayText) {
    switch (messageType) {
        case SiteDetailsMessageType.FailedToLoadData:
            return displayText.failedToLoadDataTitle;
        default: return undefined;
    }
}
function getDescriptionString(messageType, displayText) {
    switch (messageType) {
        case SiteDetailsMessageType.FailedToLoadData:
            return displayText.failedToLoadDataMessage;
        default: return undefined;
    }
}
export var SiteDetailsMessage = function (_a) {
    var messageType = _a.messageType, displayText = _a.displayText, assets = _a.assets;
    return (React.createElement(Message, { icon: getIconName(messageType, assets), title: getTitleString(messageType, displayText), description: getDescriptionString(messageType, displayText), className: 'v-site-details-message' }));
};
