// @vista/web-widgets/lib/widgets/src/components/site-list/components/site-list-message.js
import * as React from 'react';
import { Message } from '../../message/message';
export var SiteListMessageType;
(function (SiteListMessageType) {
    SiteListMessageType["FailedToLoadData"] = "FailedToLoadData";
})(SiteListMessageType || (SiteListMessageType = {}));
function getIconName(messageType, assets) {
    switch (messageType) {
        case SiteListMessageType.FailedToLoadData:
            return assets.failedToLoadDataIconName;
    }
}
function getTitleString(messageType, displayText) {
    switch (messageType) {
        case SiteListMessageType.FailedToLoadData:
            return displayText.failedToLoadDataTitle;
        default: return undefined;
    }
}
function getDescriptionString(messageType, displayText) {
    switch (messageType) {
        case SiteListMessageType.FailedToLoadData:
            return displayText.failedToLoadDataMessage;
        default: return undefined;
    }
}
export var SiteListMessage = function (_a) {
    var messageType = _a.messageType, displayText = _a.displayText, assets = _a.assets;
    return (React.createElement(Message, { icon: getIconName(messageType, assets), title: getTitleString(messageType, displayText), description: getDescriptionString(messageType, displayText), className: 'v-site-list-message' }));
};
