// @vista/web-widgets/lib/widgets/src/components/order-summary/components/order-summary-message.js
import * as React from 'react';
import { Message } from '../../message/message';
export var OrderSummaryMessageType;
(function (OrderSummaryMessageType) {
    OrderSummaryMessageType["FailedToLoadData"] = "FailedToLoadData";
})(OrderSummaryMessageType || (OrderSummaryMessageType = {}));
function getIconName(messageType, assets) {
    switch (messageType) {
        case OrderSummaryMessageType.FailedToLoadData:
            return assets.failedToLoadDataIconName;
    }
}
function getTitleString(messageType, displayText) {
    switch (messageType) {
        case OrderSummaryMessageType.FailedToLoadData:
            return displayText.failedToLoadDataTitle;
        default: return undefined;
    }
}
function getDescriptionString(messageType, displayText) {
    switch (messageType) {
        case OrderSummaryMessageType.FailedToLoadData:
            return displayText.failedToLoadDataMessage;
        default: return undefined;
    }
}
export var OrderSummaryMessage = function (_a) {
    var messageType = _a.messageType, displayText = _a.displayText, assets = _a.assets;
    return (React.createElement(Message, { icon: getIconName(messageType, assets), title: getTitleString(messageType, displayText), description: getDescriptionString(messageType, displayText), className: 'v-order-summary-message' }));
};
