// @vista/web-widgets/lib/example-data/order/order-seat.js
export var exampleOrderSeatA1 = {
    position: {
        rowIndex: 1,
        columnIndex: 1,
        areaNumber: 1
    },
    rowLabel: 'A',
    seatLabel: '1',
    areaCategoryCode: 'AreaCategory1'
};
export var exampleOrderSeatA2 = {
    position: {
        rowIndex: 1,
        columnIndex: 2,
        areaNumber: 1
    },
    rowLabel: 'A',
    seatLabel: '2',
    areaCategoryCode: 'AreaCategory1'
};
export var exampleOrderSeatA3 = {
    position: {
        rowIndex: 1,
        columnIndex: 3,
        areaNumber: 1
    },
    rowLabel: 'A',
    seatLabel: '3',
    areaCategoryCode: 'AreaCategory1'
};
export var exampleOrderSeatB1 = {
    position: {
        rowIndex: 2,
        columnIndex: 1,
        areaNumber: 1
    },
    rowLabel: 'B',
    seatLabel: '1',
    areaCategoryCode: 'AreaCategory1'
};
export var exampleOrderSeatB2 = {
    position: {
        rowIndex: 2,
        columnIndex: 2,
        areaNumber: 1
    },
    rowLabel: 'B',
    seatLabel: '2',
    areaCategoryCode: 'AreaCategory1'
};
export var exampleOrderSeatB3 = {
    position: {
        rowIndex: 2,
        columnIndex: 3,
        areaNumber: 1
    },
    rowLabel: 'B',
    seatLabel: '3',
    areaCategoryCode: 'AreaCategory1'
};
export var exampleOrderSeats = [
    exampleOrderSeatA1,
    exampleOrderSeatA2,
    exampleOrderSeatA3,
    exampleOrderSeatB1,
    exampleOrderSeatB2,
    exampleOrderSeatB3
];
