// @vista/web-widgets/lib/example-data/order/order-ticket-type.js
import { TicketTypeCategory } from '../../types/sales-data/ticket-type';
export var exampleStandardTicketType1 = {
    id: '1',
    key: 'standard-ticket-type-1',
    description: 'Adult',
    descriptionTranslations: [],
    areaCategoryCode: '0000000001',
    price: 15.00,
    category: TicketTypeCategory.Standard
};
export var exampleStandardTicketType2 = {
    id: '2',
    key: 'standard-ticket-type-2',
    description: 'Child',
    descriptionTranslations: [],
    areaCategoryCode: '0000000001',
    price: 11.00,
    category: TicketTypeCategory.Standard
};
export var exampleStandardTicketType3 = {
    id: '3',
    key: 'standard-ticket-type-3',
    description: 'Senior Citizen',
    descriptionTranslations: [],
    areaCategoryCode: '0000000001',
    price: 12.50,
    category: TicketTypeCategory.Standard
};
export var exampleStandardTicketTypes = [
    exampleStandardTicketType1,
    exampleStandardTicketType2,
    exampleStandardTicketType3
];
export var exampleLoyaltyTicketType1 = {
    id: '1',
    key: 'loyalty-ticket-type-1-reward-5',
    description: 'Adult Loyalty',
    descriptionTranslations: [],
    areaCategoryCode: '0000000001',
    price: 10.00,
    category: TicketTypeCategory.Loyalty,
    rewardDetails: { id: '5', quantityAvailable: 10 }
};
export var exampleLoyaltyTicketType2 = {
    id: '2',
    key: 'loyalty-ticket-type-2-reward-3',
    description: 'Child Loyalty',
    descriptionTranslations: [],
    areaCategoryCode: '0000000001',
    price: 8.00,
    category: TicketTypeCategory.Loyalty,
    rewardDetails: { id: '3', quantityAvailable: 5 }
};
export var exmapleLoyaltyTicketTypes = [
    exampleLoyaltyTicketType1,
    exampleLoyaltyTicketType2
];
export var examplePackageTicketType = {
    id: '4',
    key: 'package-ticket-type-4',
    description: 'Family Pass',
    descriptionTranslations: [],
    areaCategoryCode: '0000000001',
    price: 50,
    category: TicketTypeCategory.Package,
    packageContents: {
        tickets: [
            { description: 'Child', quantity: 3, descriptionTranslations: [] },
            { description: 'Adult', quantity: 2, descriptionTranslations: [] }
        ],
        items: []
    }
};
export var exampleVoucherTicketType1 = {
    id: '5',
    key: 'voucher-ticket-type-5',
    description: 'Adult Voucher',
    descriptionTranslations: [],
    areaCategoryCode: '0000000001',
    price: 0,
    category: TicketTypeCategory.Voucher
};
export var exampleVoucherTicketType2 = {
    id: '6',
    key: 'voucher-ticket-type-6',
    description: 'Child Voucher',
    descriptionTranslations: [],
    areaCategoryCode: '0000000001',
    price: 0,
    category: TicketTypeCategory.Voucher
};
export var exampleVoucherTicketTypes = [
    exampleVoucherTicketType1,
    exampleVoucherTicketType2
];
export var exampleOrderTicketTypes = exampleStandardTicketTypes.concat(exmapleLoyaltyTicketTypes, [
    examplePackageTicketType
], exampleVoucherTicketTypes);
export var exampleOrderTicketTypesMap = new Map([
    [exampleStandardTicketType1.key, exampleStandardTicketType1],
    [exampleStandardTicketType2.key, exampleStandardTicketType2],
    [exampleStandardTicketType3.key, exampleStandardTicketType3],
    [exampleLoyaltyTicketType1.key, exampleLoyaltyTicketType1],
    [exampleLoyaltyTicketType2.key, exampleLoyaltyTicketType2],
    [examplePackageTicketType.key, examplePackageTicketType],
    [exampleVoucherTicketType1.key, exampleVoucherTicketType1],
    [exampleVoucherTicketType2.key, exampleVoucherTicketType2],
]);
