// @vista/web-widgets/lib/example-data/order/order-showtime.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { exampleFilm } from '../browsing/film';
import { exampleOrderSeats } from './order-seat';
import { exampleStandardTicket1, exampleVoucherTicket1 } from './order-ticket';
import { exampleOrderTicketTypesMap } from './order-ticket-type';
export var exampleOrderShowtime = {
    showtimeId: '0000001',
    film: exampleFilm,
    screenName: 'Screen 1',
    startTime: new Date(2017, 10, 24),
    seats: exampleOrderSeats,
    tickets: [
        exampleVoucherTicket1,
        __assign({}, exampleStandardTicket1, { quantity: 2 })
    ],
    ticketTypes: exampleOrderTicketTypesMap,
    isAllocatedSeating: true
};
