// @vista/web-widgets/lib/example-data/order/order-ticket.js
import { OrderTicketCategory } from '../../types/order/order-ticket';
import { exampleLoyaltyTicketType1, exampleLoyaltyTicketType2, examplePackageTicketType, exampleStandardTicketType1, exampleStandardTicketType2, exampleStandardTicketType3, exampleVoucherTicketType1 } from './order-ticket-type';
export var exampleStandardTicket1 = {
    ticketTypeKey: exampleStandardTicketType1.key,
    quantity: 0,
    category: OrderTicketCategory.Standard
};
export var exampleStandardTicket2 = {
    ticketTypeKey: exampleStandardTicketType2.key,
    quantity: 0,
    category: OrderTicketCategory.Standard
};
export var exampleStandardTicket3 = {
    ticketTypeKey: exampleStandardTicketType3.key,
    quantity: 0,
    category: OrderTicketCategory.Standard
};
export var exampleLoyaltyTicket1 = {
    ticketTypeKey: exampleLoyaltyTicketType1.key,
    quantity: 0,
    category: OrderTicketCategory.Standard
};
export var exampleLoyaltyTicket2 = {
    ticketTypeKey: exampleLoyaltyTicketType2.key,
    quantity: 0,
    category: OrderTicketCategory.Standard
};
export var examplePackageTicket = {
    ticketTypeKey: examplePackageTicketType.key,
    quantity: 0,
    category: OrderTicketCategory.Standard
};
export var exampleVoucherTicket1 = {
    ticketTypeKey: exampleVoucherTicketType1.key,
    quantity: 0,
    barcode: '12345678',
    category: OrderTicketCategory.Voucher
};
export var exampleOrderTickets = [
    exampleStandardTicket1,
    exampleStandardTicket2,
    exampleStandardTicket3,
    exampleLoyaltyTicket1,
    exampleLoyaltyTicket2,
    examplePackageTicket,
    exampleVoucherTicket1
];
