// @vista/web-widgets/lib/widgets/src/components/order-payment-form/example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { orderStoreInstance } from '../../../../api/src/stores/order/order-store';
import { exampleOrder } from '../../../../example-data/order/order';
// import '../../scss/themes/dark/order-payment-form.scss';
import '../../scss/themes/light/order-payment-form.scss';
import { OrderPaymentForm } from './order-payment-form';
import { defaultOrderPaymentFormAssets } from './types/assets';
import { defaultOrderPaymentFormDisplayText } from './types/display-text';
orderStoreInstance.actions.setOrderData(exampleOrder);
var orderPaymentFormProps = {
    orderStore: orderStoreInstance,
    displayText: defaultOrderPaymentFormDisplayText,
    assets: defaultOrderPaymentFormAssets,
    events: {
        onPaymentFailed: function (error) { console.log('Payment failed', error); },
        onPaymentSuccessful: function (orderCompletionStatus) { console.log('Payment successful', orderCompletionStatus); }
    }
};
var OrderPaymentFormExample = /** @class */ (function (_super) {
    __extends(OrderPaymentFormExample, _super);
    function OrderPaymentFormExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    OrderPaymentFormExample.prototype.render = function () {
        return (React.createElement("div", { style: { display: 'block', margin: '30px auto', maxWidth: 1000 } },
            React.createElement("h1", null, "Order payment form"),
            React.createElement(OrderPaymentForm, __assign({}, orderPaymentFormProps))));
    };
    return OrderPaymentFormExample;
}(React.Component));
export { OrderPaymentFormExample };
