// @vista/web-widgets/lib/widgets/src/components/gift-card-checker/gift-card-checker.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import * as React from 'react';
import { Button, ButtonColor, ButtonSize } from '../button/button';
import { GiftCardCheckerCardDetails } from './components/gift-card-checker-card-details';
import { GiftCardCheckerForm } from './components/gift-card-checker-form';
export var GiftCardCheckerDisplayMode;
(function (GiftCardCheckerDisplayMode) {
    GiftCardCheckerDisplayMode[GiftCardCheckerDisplayMode["Form"] = 0] = "Form";
    GiftCardCheckerDisplayMode[GiftCardCheckerDisplayMode["Details"] = 1] = "Details";
})(GiftCardCheckerDisplayMode || (GiftCardCheckerDisplayMode = {}));
/**
 * Renders a component that allows the user to input a gift card number to retrieve details about the card.
 */
var GiftCardChecker = /** @class */ (function (_super) {
    __extends(GiftCardChecker, _super);
    function GiftCardChecker(props) {
        var _this = _super.call(this, props) || this;
        _this.setCardDetails = function (cardDetails) {
            _this.setState({
                cardDetails: cardDetails,
                displayMode: cardDetails ? GiftCardCheckerDisplayMode.Details : GiftCardCheckerDisplayMode.Form
            });
        };
        _this.clearCardDetails = function () {
            _this.setCardDetails(undefined);
        };
        _this.state = {
            cardDetails: undefined,
            displayMode: GiftCardCheckerDisplayMode.Form
        };
        return _this;
    }
    GiftCardChecker.prototype.render = function () {
        var _a = this.props, displayText = _a.displayText, getGiftCardDetails = _a.getGiftCardDetails, config = _a.config, captchaConfig = _a.captchaConfig;
        return (React.createElement("div", { className: "v-gift-card-checker" },
            this.state.displayMode === GiftCardCheckerDisplayMode.Form || !this.state.cardDetails
                ?
                    React.createElement(GiftCardCheckerForm, { displayText: displayText, getGiftCardDetails: getGiftCardDetails, onCardDetailsRetrieved: this.setCardDetails, config: config, captchaConfig: captchaConfig })
                :
                    React.createElement(GiftCardCheckerCardDetails, { displayText: displayText, cardDetails: this.state.cardDetails }),
            this.state.displayMode === GiftCardCheckerDisplayMode.Details &&
                React.createElement(Button, { color: ButtonColor.Primary, size: ButtonSize.Medium, label: displayText.buttonBackLabel, onClick: this.clearCardDetails })));
    };
    return GiftCardChecker;
}(React.Component));
export { GiftCardChecker };
