// @vista/web-widgets/lib/api/src/mappers/browsing/tests/browsing-test-data.js
import { parse } from 'date-fns';
import { MxFilmCategory } from '../../../../../mx-api-client/src/mxs/types/browsing/film-category';
import { FilmCategory } from '../../../../../types/browsing/film-category';
export var mxCensorRating = {
    id: 'pgr',
    classification: 'pgr',
    classificationTranslations: [],
    description: 'Parental Guidance',
    descriptionTranslations: []
};
export var widgetCensorRating = {
    id: 'pgr',
    classification: 'pgr',
    classificationTranslations: [],
    description: 'Parental Guidance',
    descriptionTranslations: []
};
export var mxFilm = {
    id: '001',
    title: 'Lego Movie',
    titleTranslations: [],
    plot: 'Legos come to life',
    plotTranslations: [],
    classificationNote: 'Suitable for all audiences',
    classificationNoteTranslations: [],
    censorRatingId: 'G',
    releaseDate: new Date('2016-05-05'),
    runtime: 95,
    trailerUrl: undefined,
    displayPriority: 1,
    actorIds: ['1'],
    directorIds: ['2'],
    genreIds: ['3'],
    categories: [MxFilmCategory.NowShowing]
};
export var widgetFilm = {
    id: '001',
    title: 'Lego Movie',
    titleTranslations: [],
    plot: 'Legos come to life',
    plotTranslations: [],
    classificationNote: 'Suitable for all audiences',
    classificationNoteTranslations: [],
    censorRatingId: 'G',
    releaseDate: new Date('2016-05-05'),
    runtime: 95,
    trailerUrl: undefined,
    displayPriority: 1,
    actorIds: ['1'],
    directorIds: ['2'],
    genreIds: ['3'],
    categories: [FilmCategory.NowShowing]
};
export var mxGenre = {
    id: 'd000',
    name: 'Drama',
    nameTranslations: [],
    description: 'Drama films',
    descriptionTranslations: []
};
export var widgetGenre = {
    id: 'd000',
    name: 'Drama',
    nameTranslations: [],
    description: 'Drama films',
    descriptionTranslations: []
};
export var mxPerson = {
    id: 'person01',
    givenName: 'Bradley',
    familyName: 'Cooper'
};
export var widgetPerson = {
    id: 'person01',
    givenName: 'Bradley',
    familyName: 'Cooper'
};
export var mxScreeningDetail = {
    filmId: '002',
    siteIds: ['1', '3', '5']
};
export var mxScreeningDate = {
    businessDate: '2018-02-02',
    screeningDetails: [mxScreeningDetail]
};
export var widgetScreeningDetail = {
    filmId: '002',
    siteIds: ['1', '3', '5']
};
export var widgetScreeningDate = {
    businessDate: '2018-02-02',
    screeningDetails: [mxScreeningDetail]
};
export var mxShowtimeAttribute = {
    id: '123',
    shortName: 'IMAX',
    shortNameTranslations: [],
    description: 'IMAX with Laser',
    descriptionTranslations: [],
    displayPriority: 2,
    isPromoted: false
};
export var widgetShowtimeAttribute = {
    id: '123',
    shortName: 'IMAX',
    shortNameTranslations: [],
    description: 'IMAX with Laser',
    descriptionTranslations: [],
    displayPriority: 2,
    isPromoted: false
};
export var mxShowtime = {
    id: '123',
    filmId: 'FILM001',
    siteId: 'SITE2',
    startTime: '2020-01-01T12:15:00',
    businessDate: '2020-01-01',
    attributeIds: ['IMAX'],
    isAllocatedSeating: true,
    screenName: 'IMAX 1',
    screenNameTranslations: []
};
export var widgetShowtime = {
    id: '123',
    filmId: 'FILM001',
    siteId: 'SITE2',
    startTime: parse('2020-01-01T12:15:00'),
    businessDate: '2020-01-01',
    attributeIds: ['IMAX'],
    isAllocatedSeating: true,
    screenName: 'IMAX 1',
    screenNameTranslations: []
};
export var mxSiteAddress = {
    line1: '123 Fake Street',
    line2: 'Downtown',
    city: 'Los Angeles'
};
export var widgetSiteAddress = {
    line1: '123 Fake Street',
    line2: 'Downtown',
    city: 'Los Angeles'
};
export var mxSiteContactDetails = {
    phoneNumber1: '555 0125',
    phoneNumber2: '0800 000000',
    faxNumber: '1900 123456',
    emailAddress: 'stlukes@abccinemas.co'
};
export var widgetSiteContactDetails = {
    phoneNumber1: '555 0125',
    phoneNumber2: '0800 000000',
    faxNumber: '1900 123456',
    emailAddress: 'stlukes@abccinemas.co'
};
export var mxSiteLocation = {
    longitude: -36.894655,
    latitude: 174.775372
};
export var widgetSiteLocation = {
    longitude: -36.894655,
    latitude: 174.775372
};
export var mxSite = {
    id: 'site1',
    name: 'ABC Downtown',
    nameTranslations: [],
    address: mxSiteAddress,
    location: mxSiteLocation,
    contactDetails: mxSiteContactDetails
};
export var widgetSite = {
    id: 'site1',
    name: 'ABC Downtown',
    nameTranslations: [],
    address: widgetSiteAddress,
    location: widgetSiteLocation,
    contactDetails: widgetSiteContactDetails
};
