// @vista/web-widgets/lib/types/stores/configuration-store-data.js
import { CdnReferenceScheme } from '@vista/cdn-client';
import { defaultCurrencyConfiguration, defaultDateConfiguration } from '@vista/i18n';
var ConfigurationStoreData = /** @class */ (function () {
    function ConfigurationStoreData() {
    }
    return ConfigurationStoreData;
}());
export { ConfigurationStoreData };
export var defaultConfigurationStoreData = {
    api: {
        url: '',
        authToken: '',
        cinemaChainId: 'default',
    },
    cdn: {
        url: '',
        mediaApiToken: undefined,
        referenceScheme: CdnReferenceScheme.Cinema
    },
    ticketing: {
        isSeatFirstOrdering: false,
        maximumAllowedTicketsInAnOrder: 20,
        admittingDuration: 1
    },
    browsing: {
        maximumSelectedSites: 5
    },
    localisation: {
        languageTag: 'en',
        currency: defaultCurrencyConfiguration,
        date: defaultDateConfiguration,
        filmPosterAltTextFormat: 'Theatrical poster for ${filmTitle}',
    },
    security: {
        captcha: {}
    },
    customerDetails: {
        validation: {
            email: {
                format: 'email',
                required: true,
                minLength: 0,
                maxLength: 100,
            },
            familyName: {
                format: '',
                required: true,
                minLength: 0,
                maxLength: 100,
            },
            givenName: {
                format: '',
                required: true,
                minLength: 0,
                maxLength: 100,
            },
            phone: {
                format: '',
                required: true,
                minLength: 0,
                maxLength: 25,
            },
        },
    },
    cache: {
        durations: {
            showtimes: 30,
            screeningDates: 60,
            itemButtonGroups: 60
        }
    },
    enableLoadingIndicator: true
};
