// @vista/web-widgets/lib/types/stores/order-store-data.js
import { defaultOrderData } from '../order/order';
import { defaultOrderCustomerDetails } from './../order/order-customer-details';
export var defaultOrderStoreData = {
    order: defaultOrderData,
    customerDetails: defaultOrderCustomerDetails,
    ticketSelectionErrors: [],
    customerDetailsErrors: {
        isValid: false,
        hasBeenValidated: false,
        validationErrors: new Map(),
    }
};
export var TicketSelectionValidationError;
(function (TicketSelectionValidationError) {
    TicketSelectionValidationError[TicketSelectionValidationError["NoTicketsSelected"] = 0] = "NoTicketsSelected";
    TicketSelectionValidationError[TicketSelectionValidationError["TooManyTickets"] = 1] = "TooManyTickets";
    TicketSelectionValidationError[TicketSelectionValidationError["TooFewTickets"] = 2] = "TooFewTickets";
    TicketSelectionValidationError[TicketSelectionValidationError["TicketsDoNotMatchSelectedSeats"] = 3] = "TicketsDoNotMatchSelectedSeats";
})(TicketSelectionValidationError || (TicketSelectionValidationError = {}));
