// @vista/web-widgets/lib/widgets/src/components/order-summary/bootstrap.js
import { addWidgetBootstrap } from '../../bootstrap/add-widget-bootstrap';
import { createComponent } from '../../bootstrap/create-component';
import { journeyStoreInstance } from './../../../../api/src/stores/journey/journey-store';
import { mapOrderSummaryAssets } from './mappers/assets-mapper';
import { mapOrderSummaryDisplayText } from './mappers/display-text-mapper';
import { OrderSummary } from './order-summary';
export var orderSummary = function (bootstrap) {
    var journeyReference = bootstrap.journeyReference, journeyId = bootstrap.journeyId;
    if (!journeyReference && !journeyId) {
        throw new Error('Cannot render the Order Summary widget without a journeyReference or journeyId');
    }
    var props = {
        journeyId: journeyId,
        journeyReference: journeyReference,
        journeyStore: journeyStoreInstance,
        assets: mapOrderSummaryAssets(bootstrap.assets),
        displayText: mapOrderSummaryDisplayText(bootstrap.displayText)
    };
    return createComponent(bootstrap.selector, OrderSummary, props);
};
addWidgetBootstrap('orderSummary', orderSummary);
