// @vista/web-widgets/lib/widgets/src/components/order-payment-form/components/order-payment-form-message.js
import * as React from 'react';
import { Message } from '../../message/message';
export var OrderPaymentFormMessageType;
(function (OrderPaymentFormMessageType) {
    OrderPaymentFormMessageType["PaymentFailed"] = "PaymentFailed";
    OrderPaymentFormMessageType["FailedToInitialiseConnector"] = "FailedToInitialiseConnector";
})(OrderPaymentFormMessageType || (OrderPaymentFormMessageType = {}));
function getIconName(messageType, assets) {
    switch (messageType) {
        case OrderPaymentFormMessageType.PaymentFailed:
            return assets.paymentFailedIconName;
        case OrderPaymentFormMessageType.FailedToInitialiseConnector:
            return assets.failedToInitialiseConnectorIconName;
    }
}
function getTitleString(messageType, displayText) {
    switch (messageType) {
        case OrderPaymentFormMessageType.PaymentFailed:
            return displayText.paymentFailedTitle;
        case OrderPaymentFormMessageType.FailedToInitialiseConnector:
            return displayText.failedToInitialiseConnectorTitle;
        default: return undefined;
    }
}
function getDescriptionString(messageType, displayText) {
    switch (messageType) {
        case OrderPaymentFormMessageType.PaymentFailed:
            return displayText.paymentFailedMessage;
        case OrderPaymentFormMessageType.FailedToInitialiseConnector:
            return displayText.failedToInitialiseConnectorMessage;
        default: return undefined;
    }
}
export var OrderPaymentFormMessage = function (_a) {
    var messageType = _a.messageType, displayText = _a.displayText, assets = _a.assets;
    return (React.createElement(Message, { icon: getIconName(messageType, assets), title: getTitleString(messageType, displayText), description: getDescriptionString(messageType, displayText), className: 'v-order-payment-form-message' }));
};
