// @vista/web-widgets/lib/api/src/stores/browsing/selectors/selectors.js
import { getAllFilms } from './get-all-films';
import { getAllSites } from './get-all-sites';
import { getCensorRatingById } from './get-censor-rating-by-id';
import { getFilmById } from './get-film-by-id';
import { getFilmsForSite } from './get-films-for-site';
import { getGenresByIds } from './get-genres-by-ids';
import { getPeopleByIds } from './get-people-by-ids';
import { getScreeningDatesForQuery } from './get-screening-dates-for-query';
import { getSelectedSites } from './get-selected-sites';
import { getShowtimeAttributeById } from './get-showtime-attribute-by-id';
import { getShowtimesForQueryOnBusinessDate } from './get-showtimes-for-query-on-date';
import { getSiteById } from './get-site-by-id';
import { getUnselectedSites } from './get-unselected-sites';
import { hasMaximumSelectedSites } from './has-maximum-selected-sites';
export var browsingStoreSelectors = {
    getAllFilms: getAllFilms,
    getFilmById: getFilmById,
    getFilmsForSite: getFilmsForSite,
    getShowtimeAttributeById: getShowtimeAttributeById,
    getShowtimesForQueryOnBusinessDate: getShowtimesForQueryOnBusinessDate,
    getSiteById: getSiteById,
    getAllSites: getAllSites,
    getSelectedSites: getSelectedSites,
    getUnselectedSites: getUnselectedSites,
    hasMaximumSelectedSites: hasMaximumSelectedSites,
    getCensorRatingById: getCensorRatingById,
    getGenresByIds: getGenresByIds,
    getPeopleByIds: getPeopleByIds,
    getScreeningDatesForQuery: getScreeningDatesForQuery
};
