// @vista/web-widgets/lib/api/src/stores/browsing/actions/actions.js
import { addCensorRatings } from './add-censor-ratings';
import { addFilms } from './add-films';
import { addGenres } from './add-genres';
import { addPeople } from './add-people';
import { addSelectedSite } from './add-selected-site';
import { addShowtimeAttributes } from './add-showtime-attributes';
import { addShowtimes } from './add-showtimes';
import { addSites } from './add-sites';
import { removeSelectedSite } from './remove-selected-site';
import { setFilms } from './set-films';
import { setFilmsForSite as setFilmsForSite } from './set-films-for-site';
import { setScreeningDatesForQuery } from './set-screening-dates-for-query';
import { setShowtimesForQueryOnBusinessDate } from './set-showtimes-for-query-on-date';
import { setSites } from './set-sites';
export var browsingStoreActions = {
    setShowtimesForQueryOnBusinessDate: setShowtimesForQueryOnBusinessDate,
    addShowtimes: addShowtimes,
    addFilms: addFilms,
    setFilmsForSite: setFilmsForSite,
    setFilms: setFilms,
    addSites: addSites,
    setSites: setSites,
    addSelectedSite: addSelectedSite,
    removeSelectedSite: removeSelectedSite,
    addPeople: addPeople,
    addGenres: addGenres,
    addCensorRatings: addCensorRatings,
    addShowtimeAttributes: addShowtimeAttributes,
    setScreeningDatesForQuery: setScreeningDatesForQuery
};
