// @vista/web-widgets/lib/widgets/src/components/member-transaction-history/bootstrap.js
import { loyaltyMemberService } from '../../../../api/src/services/loyalty/loyalty-member-service';
import { configurationStoreInstance } from '../../../../api/src/stores/configuration/configuration-store';
import { addWidgetBootstrap } from '../../bootstrap/add-widget-bootstrap';
import { createComponent } from '../../bootstrap/create-component';
import { mapMemberTransactionHistoryAssets } from './mappers/assets-mapper';
import { mapMemberTransactionHistoryConfig } from './mappers/config-mapper';
import { mapMemberTransactionHistoryDisplayText } from './mappers/display-text-mapper';
import { MemberTransactionHistory } from './member-transaction-history';
/**
 * Renders the Member Transaction History widget.
 * @param bootstrap
 */
export var memberTransactionHistory = function (bootstrap) {
    var selector = bootstrap.selector, config = bootstrap.config, displayText = bootstrap.displayText, assets = bootstrap.assets;
    var props = {
        loyaltyMemberService: loyaltyMemberService,
        configurationStore: configurationStoreInstance,
        config: mapMemberTransactionHistoryConfig(config),
        displayText: mapMemberTransactionHistoryDisplayText(displayText),
        assets: mapMemberTransactionHistoryAssets(assets)
    };
    return createComponent(selector, MemberTransactionHistory, props);
};
addWidgetBootstrap('memberTransactionHistory', memberTransactionHistory);
