// @vista/web-widgets/lib/widgets/src/components/member-transaction-history/example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
// import '../../scss/themes/dark/member-transaction-history.scss';
import * as React from 'react';
import { configurationStoreInstance } from '../../../../api/src/stores/configuration/configuration-store';
import { exampleLoyaltyMemberService } from '../../../../example-data/loyalty/member-transactions/loyalty-member-service';
import '../../scss/themes/light/member-transaction-history.scss';
import '../../scss/components/message/message.scss';
import { MemberTransactionHistoryMessage, MemberTransactionHistoryMessageType } from './components/member-transaction-history-message';
import { MemberTransactionHistorySkeleton } from './components/member-transaction-history-skeleton';
import { MemberTransactionHistory } from './member-transaction-history';
import { defaultMemberTransactionHistoryAssets } from './types/assets';
import { defaultMemberTransactionHistoryConfig } from './types/config';
import { defaultMemberTransactionHistoryDisplayText } from './types/display-text';
var exampleProps = {
    loyaltyMemberService: exampleLoyaltyMemberService,
    configurationStore: configurationStoreInstance,
    config: defaultMemberTransactionHistoryConfig,
    displayText: defaultMemberTransactionHistoryDisplayText,
    assets: defaultMemberTransactionHistoryAssets
};
var exampleMessageProps = {
    messageType: MemberTransactionHistoryMessageType.NoMemberTransactionData,
    displayText: defaultMemberTransactionHistoryDisplayText,
    assets: defaultMemberTransactionHistoryAssets
};
var MemberTransactionHistoryExample = /** @class */ (function (_super) {
    __extends(MemberTransactionHistoryExample, _super);
    function MemberTransactionHistoryExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MemberTransactionHistoryExample.prototype.render = function () {
        return (React.createElement("div", null,
            React.createElement("h1", null, "Transaction history"),
            React.createElement(MemberTransactionHistory, __assign({}, exampleProps)),
            React.createElement("br", null),
            React.createElement("h2", null, "Skeleton"),
            React.createElement(MemberTransactionHistorySkeleton, null),
            React.createElement("h2", null, "Messages"),
            React.createElement(MemberTransactionHistoryMessage, __assign({}, exampleMessageProps))));
    };
    return MemberTransactionHistoryExample;
}(React.Component));
export { MemberTransactionHistoryExample };
