// @vista/web-widgets/lib/api/src/stores/seating/helpers/test/area-category-builder.js
import { AreaBuilder } from './area-builder';
var AreaCategoryBuilder = /** @class */ (function () {
    function AreaCategoryBuilder(areaCategory) {
        this.areas = [];
        this.areaCategory = areaCategory;
    }
    AreaCategoryBuilder.prototype.addArea = function (areaBuilderFn) {
        var area = {
            number: this.areas.length + 1,
            areaCategoryCode: this.areaCategory.areaCategoryCode,
            columnCount: 0,
            rowCount: 0,
            rows: []
        };
        this.areas.push(area);
        var areaBuilder = new AreaBuilder(this.areaCategory, area);
        areaBuilderFn(areaBuilder);
        return this;
    };
    AreaCategoryBuilder.prototype.setTotalSeatsToSelect = function (totalSeatsToSelect) {
        this.areaCategory.seatsNotAllocatedCount = totalSeatsToSelect - this.areaCategory.seatsAllocatedCount;
        this.areaCategory.seatsToAllocate = totalSeatsToSelect;
    };
    return AreaCategoryBuilder;
}());
export { AreaCategoryBuilder };
