// @vista/web-widgets/lib/api/src/stores/seating/helpers/test/seat-layout-builder.js
import { AreaCategoryBuilder } from './area-category-builder';
var SeatLayoutBuilder = /** @class */ (function () {
    function SeatLayoutBuilder() {
        this.areaCategories = [];
        this.areas = [];
    }
    SeatLayoutBuilder.prototype.addAreaCategory = function (areaCategoryBuilderFn) {
        var areaCategory = {
            areaCategoryCode: (this.areaCategories.length + 1).toString(),
            seatsToAllocate: 0,
            seatsAllocatedCount: 0,
            seatsNotAllocatedCount: 0,
            selectedSeats: []
        };
        this.areaCategories.push(areaCategory);
        var areaCategoryBuilder = new AreaCategoryBuilder(areaCategory);
        areaCategoryBuilderFn(areaCategoryBuilder);
        this.areas = this.areas.concat(areaCategoryBuilder.areas);
        return this;
    };
    SeatLayoutBuilder.prototype.create = function () {
        return {
            areaCategories: this.areaCategories,
            areas: this.areas
        };
    };
    return SeatLayoutBuilder;
}());
export { SeatLayoutBuilder };
