// @vista/web-widgets/lib/widgets/src/components/showtime-picker/types/display-text.js
/**
 * Default display text for the Showtime Picker widget.
 */
export var defaultShowtimePickerDisplayText = {
    showtimeTimeFormat: 'h:mm',
    showtimeAmPmFormat: 'A',
    commonAttributesListLabel: 'All Showtimes:',
    runtimeFormat: '${runtimeMinutes} minutes',
    noSitesSelectedTitle: 'No cinemas selected.',
    noSitesSelectedMessage: 'Where would you like to see the movie?',
    noShowtimesForSiteOnDateMessage: 'There are no showtimes at this location on this date.',
    noShowtimesForFilmMessage: 'There are no showtimes on this date for “${selectedFilmTitle}” at your selected cinemas. \nPlease select another date or cinema.',
    noShowtimesGenericMessage: 'There are no showtimes on this date at your selected cinemas. \nPlease select another date or cinema.',
    actionButtonLabel: 'Add cinemas',
    noShowtimesForAnyDateTitle: 'No showtimes available.',
    noShowtimesForAnyDateMessage: 'Please try another selection.',
    failedToLoadDataTitle: 'Sorry, something went wrong.',
    failedToLoadDataMessage: 'Refresh your browser to try loading your selections again.'
};
