// @vista/web-widgets/lib/widgets/src/components/ticket-picker/components/ticket-picker-message.js
import * as React from 'react';
import { Message } from '../../message/message';
export var TicketPickerMessageType;
(function (TicketPickerMessageType) {
    TicketPickerMessageType["FailedToLoadData"] = "FailedToLoadData";
})(TicketPickerMessageType || (TicketPickerMessageType = {}));
function getIconName(messageType, assets) {
    switch (messageType) {
        case TicketPickerMessageType.FailedToLoadData:
            return assets.failedToLoadDataIconName;
    }
}
function getTitleString(messageType, displayText) {
    switch (messageType) {
        case TicketPickerMessageType.FailedToLoadData:
            return displayText.failedToLoadDataTitle;
        default: return undefined;
    }
}
function getDescriptionString(messageType, displayText) {
    switch (messageType) {
        case TicketPickerMessageType.FailedToLoadData:
            return displayText.failedToLoadDataMessage;
        default: return undefined;
    }
}
export var TicketPickerMessage = function (_a) {
    var messageType = _a.messageType, displayText = _a.displayText, assets = _a.assets;
    return (React.createElement(Message, { icon: getIconName(messageType, assets), title: getTitleString(messageType, displayText), description: getDescriptionString(messageType, displayText), className: 'v-ticket-picker-message' }));
};
