// @vista/web-widgets/lib/widgets/src/components/carousel/example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import '../../scss/components/carousel/carousel.scss';
import { Carousel } from './carousel';
var testProps = {
    settings: {
        slidesToShow: 2.5,
        slidesToScroll: 2,
        duration: 0.5,
        draggable: true,
        responsive: [
            {
                breakpoint: 768,
                settings: {
                    slidesToShow: 5.5,
                    slidesToScroll: 5,
                }
            }
        ]
    }
};
var CarouselExample = /** @class */ (function (_super) {
    __extends(CarouselExample, _super);
    function CarouselExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CarouselExample.prototype.render = function () {
        return (React.createElement("div", null,
            React.createElement(Carousel, __assign({}, testProps), new Array(10).fill(React.createElement("div", { style: { backgroundColor: 'pink', border: '1px solid green', minHeight: '200px' } })))));
    };
    return CarouselExample;
}(React.Component));
export { CarouselExample };
