// @vista/web-widgets/lib/widgets/src/components/seat-picker/types/config.js
import { NotificationStyle } from '../../notification/notification';
/*
* Positioning options for the Seat Picker screen.
*/
export var SeatPickerWidgetScreenPosition;
(function (SeatPickerWidgetScreenPosition) {
    SeatPickerWidgetScreenPosition["Top"] = "Top";
    SeatPickerWidgetScreenPosition["Bottom"] = "Bottom";
})(SeatPickerWidgetScreenPosition || (SeatPickerWidgetScreenPosition = {}));
export var SeatPickerWidgetShape;
(function (SeatPickerWidgetShape) {
    SeatPickerWidgetShape["Round"] = "Round";
    SeatPickerWidgetShape["Rectangle"] = "Rectangle";
})(SeatPickerWidgetShape || (SeatPickerWidgetShape = {}));
/*
 * Positioning options for the Seat Picker widget's legend.
 */
export var SeatPickerWidgetLegendPosition;
(function (SeatPickerWidgetLegendPosition) {
    SeatPickerWidgetLegendPosition["Top"] = "Top";
    SeatPickerWidgetLegendPosition["Bottom"] = "Bottom";
})(SeatPickerWidgetLegendPosition || (SeatPickerWidgetLegendPosition = {}));
/*
* Default configuration options for the Seat Picker widget.
*/
export var defaultSeatPickerConfig = {
    screen: {
        height: 35,
        margin: 10,
        bevelOffset: 30,
        position: SeatPickerWidgetScreenPosition.Top
    },
    seats: {
        minimumSize: 44,
        iconSize: 55,
        icons: {
            wheelchair: 'wheelchair',
            companion: 'person',
            friend: 'star'
        },
        margin: 4,
        shape: SeatPickerWidgetShape.Rectangle,
        borderRadius: 4,
        borderWidth: 2,
        showLabels: false
    },
    rowLabels: {
        size: 25,
        margin: 8,
        shape: SeatPickerWidgetShape.Rectangle,
        borderRadius: 3,
        borderWidth: 0
    },
    areaLabels: {
        margin: 10,
        show: false
    },
    legend: {
        position: SeatPickerWidgetLegendPosition.Top,
        items: {
            available: {
                show: true,
                index: 1
            },
            unavailable: {
                show: true,
                index: 2
            },
            selected: {
                show: true,
                index: 3
            },
            wheelchair: {
                show: true,
                index: 4,
                icon: 'wheelchair'
            },
            companion: {
                show: true,
                index: 5,
                icon: 'person'
            },
            friend: {
                show: false,
                index: 6,
                icon: 'star'
            }
        },
        seatSize: 25,
        seatIconSize: 70,
        seatShape: SeatPickerWidgetShape.Rectangle,
        seatBorderRadius: 4,
        seatBorderWidth: 2
    },
    notification: {
        show: true,
        style: NotificationStyle.Warning
    },
    navigation: {
        showPanButtons: false,
        showZoomButtons: true
    }
};
