// @vista/web-widgets/lib/widgets/src/components/seat-picker/types/display-text.js
/*
* Default display text for the Seat Picker widget.
*/
export var defaultSeatPickerDisplayText = {
    screenLabel: 'screen',
    legendItemAvailableLabel: 'Available',
    legendItemUnavailableLabel: 'Unavailable',
    legendItemSelectedLabel: 'Your Seat',
    legendItemWheelchairLabel: 'Wheelchair',
    legendItemCompanionLabel: 'Companion',
    legendItemFriendLabel: 'Friend',
    errorExceededAllocatedSeatsMessage: 'You cannot exceed your allocated seats for an area category',
    errorGapBetweenSelectedSeatsMessage: 'You cannot leave a gap between seats you have selected',
    errorSingleSeatGapFromAisleMessage: 'You cannot leave a single seat gap from an aisle',
    errorSingleSeatGapFromUnavailableSeatMessage: 'You cannot leave a single seat gap from an unavailable seat',
    errorSelectedUnavailableSeatMessage: 'One or more seats you have chosen are no longer available',
    errorEntireSofaNotSelectedMessage: 'You must select the entire sofa',
    errorNotAllSeatsSelectedMessage: 'You haven’t selected all your seats',
    errorCompanionSeatNotSelectedMessage: 'You must select the associated wheelchair seat for all companion seats',
    errorNoSeatsSelectedMessage: 'You must select at least one seat',
    errorApiSelectedUnavailableSeatsMessage: 'The seats you have selected are no longer available',
    errorApiFailedToReserveSeatsMessage: 'Failed to reserve your seats',
    failedToLoadDataTitle: 'Sorry, something went wrong.',
    failedToLoadDataMessage: 'Refresh your browser to try loading seat selection again.'
};
