// @vista/web-widgets/lib/widgets/src/components/date-picker/date-picker.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import { LoadingState } from '../../../../types/loading-state';
import { deviceHasTouch } from '../../dom/device-has-touch';
import { bemClassBuilderFactory } from '../../utils/bem-class-builder';
import { Carousel } from '../carousel/carousel';
import { DatePickerSkeleton } from '../date-picker/date-picker-skeleton';
import { DatePickerDate } from './date-picker-date';
import { getDatePickerDateValues } from './utils';
var DatePicker = /** @class */ (function (_super) {
    __extends(DatePicker, _super);
    function DatePicker() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    DatePicker.prototype.render = function () {
        var _a = this.props, loadingState = _a.loadingState, businessDates = _a.businessDates, selectedBusinessDate = _a.selectedBusinessDate, onDateClicked = _a.onDateClicked;
        var b = bemClassBuilderFactory('v-date-picker');
        var values = getDatePickerDateValues(businessDates, selectedBusinessDate);
        if (loadingState === LoadingState.Failure)
            return null;
        if (loadingState === LoadingState.Loading || loadingState === LoadingState.NotLoaded) {
            return React.createElement(DatePickerSkeleton, null);
        }
        var hasTouch = deviceHasTouch();
        var carouselProps = {
            showDots: false,
            settings: {
                slidesToScroll: 'auto',
                slidesToShow: hasTouch ? 2.5 : 3,
                responsive: [
                    {
                        breakpoint: 420,
                        settings: {
                            slidesToShow: hasTouch ? 3.5 : 4
                        }
                    },
                    {
                        breakpoint: 500,
                        settings: {
                            slidesToShow: hasTouch ? 5.5 : 5
                        }
                    },
                    {
                        breakpoint: 900,
                        settings: {
                            slidesToShow: hasTouch ? 7.5 : 7
                        }
                    }
                ]
            }
        };
        return (React.createElement("div", { className: b.block() },
            React.createElement(Carousel, __assign({}, carouselProps), values.map(function (value) {
                return React.createElement(DatePickerDate, { key: value.businessDate, value: value, onClick: onDateClicked });
            }))));
    };
    return DatePicker;
}(React.Component));
export { DatePicker };
