// @vista/web-widgets/lib/api/src/stores/order/actions/actions.js
import { addItem } from './add-item';
import { addShowtime } from './add-showtime';
import { addVoucher } from './add-voucher';
import { clearCustomerDetailsErrors } from './clear-customer-details-errors';
import { clearLoyaltyTicketTypes } from './clear-loyalty-ticket-types';
import { clearOrder } from './clear-order';
import { clearTicketSelectionErrors } from './clear-ticket-selection-errors';
import { initializeOrder } from './initialize-order';
import { removeItem } from './remove-item';
import { setOrderData } from './set-order-data';
import { setSeatsForShowtime } from './set-seats-for-showtime';
import { setTicketQuantity } from './set-ticket-quantity';
import { setTicketSelectionErrors } from './set-ticket-selection-errors';
import { setTicketTypesForShowtime } from './set-ticket-types-for-showtime';
import { setTicketsForShowtime } from './set-tickets-for-showtime';
import { updateCustomerDetails } from './update-customer-details';
import { updateCustomerDetailsErrors } from './update-customer-details-errors';
import { validateSelectedTicketsForShowtime } from './validate-selected-tickets-for-showtime';
export var orderStoreActions = {
    addVoucher: addVoucher,
    updateCustomerDetails: updateCustomerDetails,
    validateSelectedTicketsForShowtime: validateSelectedTicketsForShowtime,
    addShowtime: addShowtime,
    setTicketTypesForShowtime: setTicketTypesForShowtime,
    setTicketQuantity: setTicketQuantity,
    clearLoyaltyTicketTypes: clearLoyaltyTicketTypes,
    setSeatsForShowtime: setSeatsForShowtime,
    initializeOrder: initializeOrder,
    setOrderData: setOrderData,
    setTicketsForShowtime: setTicketsForShowtime,
    setTicketSelectionErrors: setTicketSelectionErrors,
    clearTicketSelectionErrors: clearTicketSelectionErrors,
    updateCustomerDetailsErrors: updateCustomerDetailsErrors,
    clearCustomerDetailsErrors: clearCustomerDetailsErrors,
    addItem: addItem,
    removeItem: removeItem,
    clearOrder: clearOrder
};
