// @vista/cdn-client/lib/types/cdn-media-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * The media types currently supported by the Vista CDN.
 */
var CdnMediaType;
(function (CdnMediaType) {
    /**
     * An icon image for a Film/Session Attribute. For example: IMAX.
     */
    CdnMediaType["AttributeIcon"] = "AttributeIconGraphic";
    /**
     * An advertising image for a Film/Session Attribute.
     */
    CdnMediaType["AttributeAdvertisement"] = "AttributeAdvertising";
    /**
     * An image for an Advertising Profile.
     */
    CdnMediaType["AdvertisingProfileImage"] = "AdvertisingProfileItem";
    /**
     * A video for an Advertising Profile.
     */
    CdnMediaType["AdvertisingProfileVideo"] = "AdvertisingProfileVideo";
    /**
     * A gallery of images for a Site.
     */
    CdnMediaType["SiteGallery"] = "CinemaGallery";
    /**
     * An image for a Site.
     */
    CdnMediaType["SiteImage"] = "CinemaGraphic";
    /**
     * An image for a Corporate Booking.
     */
    CdnMediaType["CorporateBookingImage"] = "CorporateBookingImage";
    /**
     * A gallery of images for an Event.
     */
    CdnMediaType["EventGallery"] = "EventGallery";
    /**
     * A poster image for an Event.
     */
    CdnMediaType["EventPoster"] = "EventPosterGraphic";
    /**
     * An advertising image for a Film.
     */
    CdnMediaType["FilmAdvertisement"] = "FilmAdvertising";
    /**
     * A banner image for a Film.
     */
    CdnMediaType["FilmBackdrop"] = "FilmBackdrop";
    /**
     * An animated poster video for a Film.
     */
    CdnMediaType["FilmMotionPoster"] = "FilmMotionPoster";
    /**
     * A poster image for a Film.
     */
    CdnMediaType["FilmPoster"] = "FilmPosterGraphic";
    /**
     * A still image taken from a Film.
     */
    CdnMediaType["FilmStill"] = "FilmStill";
    /**
     * An image of a stylised Film title.
     */
    CdnMediaType["FilmTitleTreatment"] = "FilmTitleGraphic";
    /**
     * A full-length trailer for a Film.
     */
    CdnMediaType["FilmTrailer"] = "FilmTrailerFull";
    /**
     * A short/teaser trailer for a Film.
     */
    CdnMediaType["FilmShortTrailer"] = "FilmTrailerShort";
    /**
     * An image for an Item.
     */
    CdnMediaType["ItemImage"] = "ItemGraphic";
    /**
     * An advertising image for an Item.
     */
    CdnMediaType["ItemAdvertisement"] = "ItemAdvertising";
    /**
     * An image for a Promotion or Deal.
     */
    CdnMediaType["PromotionImage"] = "PromotionGraphic";
    /**
     * An advertising image for a Promotion or Deal.
     */
    CdnMediaType["PromotionAdvertisement"] = "PromotionAdvertising";
    /**
     * An icon for a Censor Rating. For example: R16.
     */
    CdnMediaType["CensorRatingIcon"] = "RatingIconGraphic";
    /**
     * An image for a Ticket Type.
     */
    CdnMediaType["TicketTypeImage"] = "TicketTypeGraphic";
    /**
     * An image for a Loyalty Club.
     */
    CdnMediaType["LoyaltyClubImage"] = "LoyaltyClubs";
    /**
     * A banner image of a Loyalty Club.
     */
    CdnMediaType["LoyaltyClubBanner"] = "LoyaltyClubBanner";
    /**
     * A login image for a Loyalty Club.
     */
    CdnMediaType["LoyaltyClubLoginImage"] = "LoyaltyClubLogin";
    /**
     * A payment image for a Loyalty Club.
     */
    CdnMediaType["LoyaltyClubPaymentImage"] = "LoyaltyClubPayment";
    /**
     * A home banner image for a Loyalty Club.
     */
    CdnMediaType["LoyaltyClubHomeBanner"] = "LoyaltyClubHomeBanner";
    /**
     * A header image for a Loyalty Club.
     */
    CdnMediaType["LoyaltyClubHeader"] = "LoyaltyClubHeader";
    /**
     * An image for a Loyalty Membership Program.
     */
    CdnMediaType["LoyaltyMembershipProgramImage"] = "LoyaltyMembershipPrograms";
    /**
     * An image for a Loyalty Reward Shop Item.
     */
    CdnMediaType["LoyaltyRewardShopItemImage"] = "LoyaltyRewardShopItems";
})(CdnMediaType = exports.CdnMediaType || (exports.CdnMediaType = {}));
