// @vista/web-widgets/lib/example-data/journey/order-ticket.js
import { JourneyOrderLineItemCollectedStatus } from '../../types/journey/order-line-item-collected-status';
import { JourneyOrderLineItemRefundedStatus } from '../../types/journey/order-line-item-refunded-status';
import { JourneyOrderTicketAdmissionStatus } from '../../types/journey/order-ticket-admission-status';
import { exampleJourneyOrderSeatA1, exampleJourneyOrderSeatA2, exampleJourneyOrderSeatA3 } from './order-seat';
export var exampleJourneyOrderTicket1 = {
    ticketTypeId: 'ticketType1',
    ticketBarcode: 'ticketBarcode1',
    orderLineNumber: 1,
    price: 9.5,
    seats: [exampleJourneyOrderSeatA1],
    collectedStatus: JourneyOrderLineItemCollectedStatus.NotCollected,
    refundedStatus: JourneyOrderLineItemRefundedStatus.NotRefunded,
    admissionStatus: JourneyOrderTicketAdmissionStatus.NotAdmitted
};
export var exampleJourneyOrderTicket2 = {
    ticketTypeId: 'ticketType2',
    ticketBarcode: 'ticketBarcode2',
    orderLineNumber: 2,
    price: 5.5,
    seats: [exampleJourneyOrderSeatA2],
    collectedStatus: JourneyOrderLineItemCollectedStatus.NotCollected,
    refundedStatus: JourneyOrderLineItemRefundedStatus.NotRefunded,
    admissionStatus: JourneyOrderTicketAdmissionStatus.NotAdmitted
};
export var exampleJourneyOrderTicket3 = {
    ticketTypeId: 'ticketType2',
    ticketBarcode: 'ticketBarcode3',
    orderLineNumber: 3,
    price: 5.5,
    seats: [exampleJourneyOrderSeatA3],
    collectedStatus: JourneyOrderLineItemCollectedStatus.NotCollected,
    refundedStatus: JourneyOrderLineItemRefundedStatus.NotRefunded,
    admissionStatus: JourneyOrderTicketAdmissionStatus.NotAdmitted
};
export var exampleJourneyOrderTickets = [
    exampleJourneyOrderTicket1,
    exampleJourneyOrderTicket2,
    exampleJourneyOrderTicket3
];
