// @vista/web-widgets/lib/example-data/journey/order-item.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { JourneyOrderLineItemCollectedStatus } from '../../types/journey/order-line-item-collected-status';
import { JourneyOrderLineItemRefundedStatus } from '../../types/journey/order-line-item-refunded-status';
export var exampleJourneyOrderItem1 = {
    itemId: 'item1',
    orderLineNumber: 4,
    price: 6,
    collectedStatus: JourneyOrderLineItemCollectedStatus.NotCollected,
    refundedStatus: JourneyOrderLineItemRefundedStatus.NotRefunded
};
export var exampleJourneyOrderItem2 = {
    itemId: 'item2',
    orderLineNumber: 5,
    price: 4.8,
    collectedStatus: JourneyOrderLineItemCollectedStatus.NotCollected,
    refundedStatus: JourneyOrderLineItemRefundedStatus.NotRefunded
};
export var exampleJourneyOrderItem3 = __assign({}, exampleJourneyOrderItem2, { orderLineNumber: 6 });
