// @vista/web-widgets/lib/example-data/journey/order-seat.js
export var exampleJourneyOrderSeatA1 = {
    seatLabel: '1',
    rowLabel: 'A',
    position: {
        areaNumber: 1,
        columnIndex: 1,
        rowIndex: 1
    }
};
export var exampleJourneyOrderSeatA2 = {
    seatLabel: '2',
    rowLabel: 'A',
    position: {
        areaNumber: 1,
        columnIndex: 2,
        rowIndex: 1
    }
};
export var exampleJourneyOrderSeatA3 = {
    seatLabel: '3',
    rowLabel: 'A',
    position: {
        areaNumber: 1,
        columnIndex: 3,
        rowIndex: 1
    }
};
export var exampleJourneyOrderSeatB1 = {
    seatLabel: '1',
    rowLabel: 'B',
    position: {
        areaNumber: 1,
        columnIndex: 1,
        rowIndex: 2
    }
};
export var exampleJourneyOrderSeatB2 = {
    seatLabel: '2',
    rowLabel: 'B',
    position: {
        areaNumber: 1,
        columnIndex: 2,
        rowIndex: 2
    }
};
export var exampleJourneyOrderSeatB3 = {
    seatLabel: '3',
    rowLabel: 'B',
    position: {
        areaNumber: 1,
        columnIndex: 3,
        rowIndex: 2
    }
};
export var exampleJourneyOrderSeats = [
    exampleJourneyOrderSeatA1,
    exampleJourneyOrderSeatA2,
    exampleJourneyOrderSeatA3,
    exampleJourneyOrderSeatB1,
    exampleJourneyOrderSeatB2,
    exampleJourneyOrderSeatB3
];
