// @vista/web-widgets/lib/widgets/src/components/site-picker/components/site-picker-message.js
import * as React from 'react';
import { Message } from '../../message/message';
export var SitePickerMessageType;
(function (SitePickerMessageType) {
    SitePickerMessageType["FailedToLoadData"] = "FailedToLoadData";
})(SitePickerMessageType || (SitePickerMessageType = {}));
function getIconName(messageType, assets) {
    switch (messageType) {
        case SitePickerMessageType.FailedToLoadData:
            return assets.failedToLoadDataIconName;
    }
}
function getTitleString(messageType, displayText) {
    switch (messageType) {
        case SitePickerMessageType.FailedToLoadData:
            return displayText.failedToLoadDataTitle;
        default: return undefined;
    }
}
function getDescriptionString(messageType, displayText) {
    switch (messageType) {
        case SitePickerMessageType.FailedToLoadData:
            return displayText.failedToLoadDataMessage;
        default: return undefined;
    }
}
export var SitePickerMessage = function (_a) {
    var messageType = _a.messageType, displayText = _a.displayText, assets = _a.assets;
    return (React.createElement(Message, { icon: getIconName(messageType, assets), title: getTitleString(messageType, displayText), description: getDescriptionString(messageType, displayText), className: 'v-site-picker-message' }));
};
