// @vista/web-widgets/lib/api/src/widgets-api/journey/utils/get-journey-order-showtime-status.js
import { journeyStoreInstance } from './../../../stores/journey/journey-store';
import { hasJourneyOrderShowtimeBeenFullyAdmitted, hasJourneyOrderShowtimeBeenFullyRefunded } from './journey-order-showtime-utils';
import { hasJourneyShowtimeEnded, isJourneyShowtimeAdmitting, isJourneyShowtimePlaying } from './journey-showtime-utils';
export var JourneyOrderShowtimeStatus;
(function (JourneyOrderShowtimeStatus) {
    JourneyOrderShowtimeStatus[JourneyOrderShowtimeStatus["Refunded"] = 0] = "Refunded";
    JourneyOrderShowtimeStatus[JourneyOrderShowtimeStatus["Pending"] = 1] = "Pending";
    JourneyOrderShowtimeStatus[JourneyOrderShowtimeStatus["Admitting"] = 2] = "Admitting";
    JourneyOrderShowtimeStatus[JourneyOrderShowtimeStatus["Admitted"] = 3] = "Admitted";
    JourneyOrderShowtimeStatus[JourneyOrderShowtimeStatus["NowPlaying"] = 4] = "NowPlaying";
    JourneyOrderShowtimeStatus[JourneyOrderShowtimeStatus["Expired"] = 5] = "Expired";
})(JourneyOrderShowtimeStatus || (JourneyOrderShowtimeStatus = {}));
/**
 * Returns the current status of a journey order showtime
 */
export function getJourneyOrderShowtimeStatus(orderShowtime) {
    var showtime = journeyStoreInstance.selectors.getShowtimeById(orderShowtime.showtimeId);
    if (!showtime)
        throw new Error("Showtime " + orderShowtime.showtimeId + " not found. Cannot get status of journey order showtime");
    var film = journeyStoreInstance.selectors.getFilmById(showtime.filmId);
    if (!film)
        throw new Error("Showtime film " + showtime.filmId + " not found. Cannot get status of journey order showtime");
    // Showtime has been refunded
    if (hasJourneyOrderShowtimeBeenFullyRefunded(orderShowtime))
        return JourneyOrderShowtimeStatus.Refunded;
    // Showtime has expired
    if (hasJourneyShowtimeEnded(showtime))
        return JourneyOrderShowtimeStatus.Expired;
    // Showtime has been collected
    if (hasJourneyOrderShowtimeBeenFullyAdmitted(orderShowtime))
        return JourneyOrderShowtimeStatus.Admitted;
    // Film has started playing
    if (isJourneyShowtimePlaying(showtime))
        return JourneyOrderShowtimeStatus.NowPlaying;
    // Film has not started playing
    if (isJourneyShowtimeAdmitting(showtime))
        return JourneyOrderShowtimeStatus.Admitting;
    // Showtime is pending
    return JourneyOrderShowtimeStatus.Pending;
}
/**
 * Returns true if a given showtime status is an active status
 */
export var isActiveJourneyOrderShowtimeStatus = function (status) {
    return status === JourneyOrderShowtimeStatus.Pending
        || status === JourneyOrderShowtimeStatus.Admitting
        || status === JourneyOrderShowtimeStatus.Admitted
        || status === JourneyOrderShowtimeStatus.NowPlaying;
};
/**
 * Returns true if a given showtime status is an inactive status
 */
export var isInactiveJourneyOrderShowtimeStatus = function (status) {
    return status === JourneyOrderShowtimeStatus.Refunded
        || status === JourneyOrderShowtimeStatus.Expired;
};
