// @vista/web-widgets/lib/api/src/widgets-api/journey/utils/get-journey-state.js
import { getJourneyOrderShowtimeStatus, isActiveJourneyOrderShowtimeStatus, JourneyOrderShowtimeStatus } from './get-journey-order-showtime-status';
import { getSortedJourneyOrderShowtimes } from './journey-utils';
/**
 * The current status of a journey.
 */
export var JourneyStatus;
(function (JourneyStatus) {
    JourneyStatus[JourneyStatus["Active"] = 0] = "Active";
    JourneyStatus[JourneyStatus["Refunded"] = 1] = "Refunded";
    JourneyStatus[JourneyStatus["Expired"] = 2] = "Expired";
})(JourneyStatus || (JourneyStatus = {}));
/**
 * Returns the current state of a given journey.
 */
export function getJourneyState(journey) {
    var sortedShowtimes = getSortedJourneyOrderShowtimes(journey);
    var nextShowtime = undefined;
    var upcomingShowtimes = [];
    var refundedShowtimes = [];
    var expiredShowtimes = [];
    sortedShowtimes.forEach(function (showtime) {
        var status = getJourneyOrderShowtimeStatus(showtime);
        if (isActiveJourneyOrderShowtimeStatus(status))
            !nextShowtime ? nextShowtime = showtime : upcomingShowtimes.push(showtime);
        else if (status === JourneyOrderShowtimeStatus.Refunded)
            refundedShowtimes.push(showtime);
        else
            expiredShowtimes.push(showtime);
    });
    var status = nextShowtime || upcomingShowtimes.length > 0 ? JourneyStatus.Active
        : refundedShowtimes.length > 0 && expiredShowtimes.length === 0 ? JourneyStatus.Refunded
            : JourneyStatus.Expired;
    return {
        status: status,
        nextOrderShowtime: nextShowtime,
        upcomingOrderShowtimes: upcomingShowtimes,
        refundedOrderShowtimes: refundedShowtimes,
        expiredOrderShowtimes: expiredShowtimes
    };
}
