// @vista/diagnostics/lib/log/helpers/client-metadata.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var uuid_1 = require("./uuid");
exports.STORAGE_KEY = 'vista-diagnostics-client-metadata';
/**
 * Returns a metadata object containing sessionId and minimal browsers informations
 */
function getClientMetadata() {
    // check if in session storage already
    var metadata = JSON.parse(sessionStorage.getItem(exports.STORAGE_KEY) || '{}');
    // if not retrieve and store in session storage
    if (!metadata.sessionId) {
        metadata = {
            sessionId: uuid_1.createUUID(),
            browser: {
                mobile: /Mobile|mini|Fennec|Android|iP(ad|od|hone)/.test(navigator.appVersion),
                userAgent: navigator.userAgent,
                resolution: screen.width + "x" + screen.height + "x" + screen.colorDepth
            }
        };
        sessionStorage.setItem(exports.STORAGE_KEY, JSON.stringify(metadata));
    }
    return metadata;
}
exports.getClientMetadata = getClientMetadata;
