// @vista/web-widgets/lib/widgets/src/components/item-picker/components/item-picker-message.js
import * as React from 'react';
import { Message } from '../../message/message';
export var ItemPickerMessageType;
(function (ItemPickerMessageType) {
    ItemPickerMessageType["FailedToLoadData"] = "FailedToLoadData";
})(ItemPickerMessageType || (ItemPickerMessageType = {}));
function getIconName(messageType, assets) {
    switch (messageType) {
        case ItemPickerMessageType.FailedToLoadData:
            return assets.failedToLoadDataIconName;
    }
}
function getTitleString(messageType, displayText) {
    switch (messageType) {
        case ItemPickerMessageType.FailedToLoadData:
            return displayText.failedToLoadDataTitle;
        default: return undefined;
    }
}
function getDescriptionString(messageType, displayText) {
    switch (messageType) {
        case ItemPickerMessageType.FailedToLoadData:
            return displayText.failedToLoadDataMessage;
        default: return undefined;
    }
}
export var ItemPickerMessage = function (_a) {
    var messageType = _a.messageType, displayText = _a.displayText, assets = _a.assets;
    return (React.createElement(Message, { icon: getIconName(messageType, assets), title: getTitleString(messageType, displayText), description: getDescriptionString(messageType, displayText), className: 'v-item-picker-message' }));
};
