// @vista/web-widgets/lib/api/src/widgets-api/journey/mappers/journey-related-data-mapper.js
import { mapMxJourneyFilm } from './journey-film-mapper';
import { mapMxJourneyItem } from './journey-item-mapper';
import { mapMxJourneyScreen } from './journey-screen-mapper';
import { mapMxJourneyShowtime } from './journey-showtime-mapper';
import { mapMxJourneySite } from './journey-site-mapper';
import { mapMxJourneyTicketType } from './journey-ticket-type-mapper';
export var mapMxJourneyRelatedData = function (relatedData) {
    var films = relatedData.films, ticketTypes = relatedData.ticketTypes, items = relatedData.items, showtimes = relatedData.showtimes, sites = relatedData.sites, screens = relatedData.screens;
    return {
        films: films && films.map(mapMxJourneyFilm),
        ticketTypes: ticketTypes && ticketTypes.map(mapMxJourneyTicketType),
        items: items && items.map(mapMxJourneyItem),
        showtimes: showtimes && showtimes.map(mapMxJourneyShowtime),
        sites: sites && sites.map(mapMxJourneySite),
        screens: screens && screens.map(mapMxJourneyScreen)
    };
};
